<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Daniel S. Himmelstein" />
  <meta name="author" content="David R. Slochower" />
  <meta name="author" content="Venkat S. Malladi" />
  <meta name="author" content="Casey S. Greene" />
  <meta name="author" content="Anthony Gitter" />
  <meta name="dcterms.date" content="2019-03-08" />
  <meta name="keywords" content="collaborative review, continuous integration, publishing, Manubot" />
  <title>Open collaborative writing with Manubot</title>
  <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <style>
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Open collaborative writing with Manubot</h1>
</header>
<p><small><em> This manuscript (<a href="https://greenelab.github.io/meta-review/v/91030dc74ebecaea7cc2ca027202f3b38892cdd3/">permalink</a>) was automatically generated from <a href="https://github.com/greenelab/meta-review/tree/91030dc74ebecaea7cc2ca027202f3b38892cdd3">greenelab/meta-review@91030dc</a> on March 8, 2019. </em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Daniel S. Himmelstein</strong><br> <img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /> <a href="https://orcid.org/0000-0002-3012-7446">0000-0002-3012-7446</a> · <img src="images/github.svg" alt="GitHub icon" class="inline_icon" /> <a href="https://github.com/dhimmel">dhimmel</a> · <img src="images/twitter.svg" alt="Twitter icon" class="inline_icon" /> <a href="https://twitter.com/dhimmel">dhimmel</a><br> <small> Department of Systems Pharmacology and Translational Therapeutics, University of Pennsylvania </small></p></li>
<li><p><strong>David R. Slochower</strong><br> <img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /> <a href="https://orcid.org/0000-0003-3928-5050">0000-0003-3928-5050</a> · <img src="images/github.svg" alt="GitHub icon" class="inline_icon" /> <a href="https://github.com/slochower">slochower</a> · <img src="images/twitter.svg" alt="Twitter icon" class="inline_icon" /> <a href="https://twitter.com/drslochower">drslochower</a><br> <small> Skaggs School of Pharmacy and Pharmaceutical Sciences, University of California, San Diego </small></p></li>
<li><p><strong>Venkat S. Malladi</strong><br> <img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /> <a href="https://orcid.org/0000-0002-0144-0564">0000-0002-0144-0564</a> · <img src="images/github.svg" alt="GitHub icon" class="inline_icon" /> <a href="https://github.com/vsmalladi">vsmalladi</a> · <img src="images/twitter.svg" alt="Twitter icon" class="inline_icon" /> <a href="https://twitter.com/katatonikkat">katatonikkat</a><br> <small> Department of Bioinformatics, University of Texas Southwestern Medical Center; Bioinformatics Core Facility, University of Texas Southwestern Medical Center </small></p></li>
<li><p><strong>Casey S. Greene</strong><br> <img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /> <a href="https://orcid.org/0000-0001-8713-9213">0000-0001-8713-9213</a> · <img src="images/github.svg" alt="GitHub icon" class="inline_icon" /> <a href="https://github.com/cgreene">cgreene</a> · <img src="images/twitter.svg" alt="Twitter icon" class="inline_icon" /> <a href="https://twitter.com/GreeneScientist">GreeneScientist</a><br> <small> Department of Systems Pharmacology and Translational Therapeutics, University of Pennsylvania </small></p></li>
<li><p><strong>Anthony Gitter</strong><br> <img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /> <a href="https://orcid.org/0000-0002-5324-9833">0000-0002-5324-9833</a> · <img src="images/github.svg" alt="GitHub icon" class="inline_icon" /> <a href="https://github.com/agitter">agitter</a> · <img src="images/twitter.svg" alt="Twitter icon" class="inline_icon" /> <a href="https://twitter.com/anthonygitter">anthonygitter</a><br> <small> Department of Biostatistics and Medical Informatics, University of Wisconsin-Madison; Morgridge Institute for Research </small></p></li>
</ul>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<p>Open, collaborative research is a powerful paradigm that can immensely strengthen the scientific process by integrating broad and diverse expertise. However, traditional research and multi-author writing processes break down at scale. New tools and workflows that rely on automation can catch errors and ensure that contributions are fairly tracked in massively collaborative research. We present techniques for overcoming challenges of open research, with special emphasis on manuscript writing. These include approaches for managing distributed authors and our new software, named Manubot, for automating citation and many other aspects of manuscript building.</p>
<h2 id="introduction" class="page_break_before">Introduction</h2>
<p>The internet enables science to be shared in real-time at a low cost to a global audience. This development has decreased the barriers to making science open, while supporting new massively collaborative models of research. However, the scientific community requires tools whose workflows encourage openness. Manuscripts are the cornerstone of scholarly communication, but drafting and publishing manuscripts has traditionally relied on proprietary or offline tools that do not support <em>open scholarly writing</em>, in which anyone is able to contribute and the contribution history is preserved and public. We introduce <a href="https://manubot.org">Manubot</a>, a new tool and infrastructure for authoring scholarly manuscripts in the open, and report how it was instrumental for the collaborative project that led to its creation.</p>
<p>Based on our experience leading a recent open review <span class="citation" data-cites="16CgM2x0z">[<a href="#ref-16CgM2x0z">1</a>]</span>, we discuss the advantages and challenges of open collaborative writing, a form of crowdsourcing <span class="citation" data-cites="12sHvZy1a">[<a href="#ref-12sHvZy1a">2</a>]</span>. Our review manuscript <span class="citation" data-cites="PZMP42Ak">[<a href="#ref-PZMP42Ak">3</a>]</span> was code-named the Deep Review and surveyed deep learning’s role in biology and precision medicine, a research area undergoing explosive growth. We initiated the Deep Review by creating a GitHub repository (<a href="https://github.com/greenelab/deep-review" class="uri">https://github.com/greenelab/deep-review</a>) to coordinate and manage contributions. GitHub is a platform designed for collaborative software development that is adaptable for collaborative writing. From the start, we made the GitHub repository public under a <a href="https://github.com/greenelab/deep-review/blob/master/LICENSE.md">Creative Commons Attribution 4.0 International (CC BY 4.0) license</a>. We encouraged anyone interested to contribute by proposing changes or additions. Although we invited some specific experts to participate, most authors discovered the manuscript organically through conferences or social media, deciding to contribute without solicitation. In total, the Deep Review attracted 36 authors, who were not determined in advance, from 20 different institutions.</p>
<p>The Deep Review and other studies that subsequently adopted the Manubot platform were unequivocal successes bolstered by the collaborative approach. However, inviting wide authorship brought many technical and social challenges such as how to fairly distribute credit, coordinate the scientific content, and collaboratively manage extensive reference lists. The manuscript writing process we developed using the Markdown language, the GitHub platform, and our new Manubot tool for automating manuscript generation addresses these challenges.</p>
<h2 id="contribution-workflow">Contribution workflow</h2>
<p>There are many existing collaborative writing platforms ranging from rich text editors, which support Microsoft Word documents or similar formats, to LaTeX-based systems for technical writing <span class="citation" data-cites="AylLD9F8">[<a href="#ref-AylLD9F8">4</a>]</span> such as <a href="https://www.overleaf.com/">Overleaf</a> and <a href="https://www.authorea.com/">Authorea</a>. These platforms ideally offer version control, multiple permission levels, or other functionality to support multi-author document editing. Although they work well for editing text, they lack sufficient features for managing a collaborative manuscript and attributing precise credit, which are important for open writing (Table <a href="#tbl:platforms">1</a>).</p>
<a name="tbl:platforms"></a>
<table style="width:100%;">
<caption><span>Table 1:</span> Collaborative writing platforms. A summary of features that differentiate Manubot from existing collaborative writing platforms. We assessed features on June 15, 2018 using the free version of each platform. Some platforms offer additional features through a paid subscription or software. <sup>1</sup>Additional functionality, such as bibliography management, is available by editing the Word document stored in OneDrive with the paid Word desktop application. <sup>2</sup>Conversations about modifications take place on the document as comments, annotations, or unsaved chats. There is no integrated forum for discussing and editing revisions. <sup>3</sup>In some circumstances, Overleaf git commits are not modular. Edits made by distinct authors may be attributed to a single author. </caption>
<colgroup>
<col style="width: 14%" />
<col style="width: 14%" />
<col style="width: 14%" />
<col style="width: 14%" />
<col style="width: 14%" />
<col style="width: 14%" />
<col style="width: 14%" />
</colgroup>
<thead>
<tr class="header">
<th>Feature</th>
<th>Manubot</th>
<th>Authorea + BibTeX</th>
<th>Overleaf v1 + BibTeX</th>
<th>Google Docs + Paperpile</th>
<th>Word Online<sup>1</sup></th>
<th>Markdown on GitHub</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Multi-author editing</td>
<td>Yes</td>
<td>Yes</td>
<td>Yes</td>
<td>Yes</td>
<td>Yes</td>
<td>Yes</td>
</tr>
<tr class="even">
<td>Propose changes</td>
<td>Yes</td>
<td>No</td>
<td>No</td>
<td>Yes</td>
<td>No</td>
<td>Yes</td>
</tr>
<tr class="odd">
<td>Continuous integration testing</td>
<td>Yes</td>
<td>No</td>
<td>No</td>
<td>No</td>
<td>No</td>
<td>No</td>
</tr>
<tr class="even">
<td>Multi-participant conversation for changes</td>
<td>Yes</td>
<td>No<sup>2</sup></td>
<td>No<sup>2</sup></td>
<td>No<sup>2</sup></td>
<td>No<sup>2</sup></td>
<td>Yes</td>
</tr>
<tr class="odd">
<td>Character-level provenance for text</td>
<td>Yes</td>
<td>No (versions tracked by day)</td>
<td>No<sup>3</sup></td>
<td>Requires manual inspection of history</td>
<td>Not after changes are accepted</td>
<td>Yes</td>
</tr>
<tr class="even">
<td>Bibliography management</td>
<td>Yes</td>
<td>Yes</td>
<td>Yes</td>
<td>Yes</td>
<td>No, requires the Word desktop application</td>
<td>No</td>
</tr>
<tr class="odd">
<td>Cite by common identifiers</td>
<td>Yes</td>
<td>No</td>
<td>No</td>
<td>No</td>
<td>No</td>
<td>No</td>
</tr>
<tr class="even">
<td>Editing software</td>
<td>Any text editor</td>
<td>Web interface</td>
<td>Web interface</td>
<td>Web interface</td>
<td>Web interface</td>
<td>Any text editor</td>
</tr>
<tr class="odd">
<td>Document format</td>
<td>Markdown</td>
<td>LaTeX</td>
<td>LaTeX</td>
<td>Proprietary</td>
<td>Proprietary</td>
<td>Markdown</td>
</tr>
<tr class="even">
<td>Templating</td>
<td>Yes</td>
<td>Yes</td>
<td>Yes</td>
<td>No</td>
<td>No</td>
<td>No</td>
</tr>
<tr class="odd">
<td>Technical expertise required</td>
<td>Yes</td>
<td>Yes</td>
<td>Yes</td>
<td>No</td>
<td>No</td>
<td>Yes</td>
</tr>
</tbody>
</table>
<p>In our workflow, we adopt standard software development strategies that enable any contributor to edit any part of the manuscript but enforce discussion and review of all proposed changes. The GitHub platform supports organizing and editing the manuscript. We use GitHub <em>issues</em> for organization, opening a new issue for each discussion topic. For example, in a review manuscript like the Deep Review, this includes each primary paper under consideration. Within a paper’s issue, contributors summarize the research, discuss it (sometimes with participation from the original authors), and assess its relevance to the review. Issues also serve as an open to-do list and a forum for debating the main message, themes, and topics of the review.</p>
<p>GitHub and the underlying git version control system <span class="citation" data-cites="PlcxShQU kEX5dgzK">[<a href="#ref-PlcxShQU">5</a>,<a href="#ref-kEX5dgzK">6</a>]</span> also structure the writing process. The official version of the manuscript is <em>forked</em> by individual contributors. A contributor then adds and revises files, grouping these changes into <em>commits</em>. When the changes are ready to be reviewed, the series of commits are submitted as a <em>pull request</em> through GitHub, which notifies other authors of the pending changes. GitHub’s review interface allows anyone to comment on the changes, globally or at specific lines, asking questions or requesting modifications, as described by Johnny Israeli <span class="citation" data-cites="opQBBK06">[<a href="#ref-opQBBK06">7</a>]</span>. Conversations during review can reference other pull requests, issues, or authors, linking the relevant people and content, as illustrated in Figure <a href="#fig:workflow">1</a>. Reviewing batches of revisions that focus on a single theme is more efficient than independently discussing isolated comments and edits and helps maintain consistent content and tone across different authors and reviewers. Once all requested modifications are made, the manuscript maintainers, a subset of authors with elevated GitHub permissions, formally approve the pull request and merge the changes into the official version. The process of writing and revising material can be orchestrated through GitHub with a web browser or a local text editor.</p>
<figure>
<img src="images/workflow.svg" alt="Figure 1: Deep Review editing workflow. Any reader can become a contributor by proposing a change through a pull request. In this example, the contributor opens an issue to discuss a manuscript modification. A maintainer and additional participant provide feedback, and the maintainer recommends creating a pull request to update the text. The contributor creates the pull request, it is reviewed by a maintainer and a participant, and the contributor updates the pull request in response. Once the pull request is approved, the maintainer merges the changes into the official version of the manuscript." id="fig:workflow" style="width:5in" /><figcaption><span>Figure 1:</span> <strong>Deep Review editing workflow.</strong> Any reader can become a contributor by proposing a change through a pull request. In this example, the contributor opens an issue to discuss a manuscript modification. A maintainer and additional participant provide feedback, and the maintainer recommends creating a pull request to update the text. The contributor creates the pull request, it is reviewed by a maintainer and a participant, and the contributor updates the pull request in response. Once the pull request is approved, the maintainer merges the changes into the official version of the manuscript.</figcaption>
</figure>
<!-- Google Drawing View Link https://docs.google.com/drawings/d/17mst0Z1RMXegeGhM6SwQSJ4nhHgMMh5csyV4MwhsgAY/edit?usp=sharing -->
<p>The Deep Review <a href="https://github.com/greenelab/deep-review/issues/575">issue</a> and <a href="https://github.com/greenelab/deep-review/pull/638">pull request</a> on protein-protein interactions demonstrate this process in practice. A new contributor identified a relevant research topic that was missing from the review manuscript with examples of how the literature would be summarized, critiqued, and integrated into the review. A maintainer confirmed that this was a desirable topic and referred to related open issues. The contributor made the pull request, and two maintainers and another participant made recommendations. After four rounds of reviews and pull request edits, a maintainer merged the changes.</p>
<p>We found that this workflow was an effective compromise between fully unrestricted editing and a more heavily-structured approach that limited the authors or the sections they could edit. In addition, authors are associated with their commits, which makes it easy for contributors to receive credit for their work and helps prevent ghostwriting <span class="citation" data-cites="RK9sIADd">[<a href="#ref-RK9sIADd">8</a>]</span>. Figure <a href="#fig:contrib">2</a> and the GitHub <a href="https://github.com/greenelab/deep-review/graphs/contributors">contributors page</a> summarize all edits and commits from each author, providing aggregated information that is not available on other collaborative writing platforms. Because the Manubot writing process tracks the complete history through git commits, it enables detailed retrospective contribution analysis.</p>
<figure>
<img src="images/deep-review-contribution-ridge.svg" alt="Figure 2: Deep Review contributions by author over time. The total words added to the Deep Review by each author is plotted over time (final values in parentheses). These statistics were extracted from git commit diffs of the manuscript’s Markdown source. This figure reveals the composition of written contributions to the manuscript at every point in its history. The Deep Review was initiated in August 2016, and the first complete manuscript was released as a preprint [9] in May 2017. While the article was under review, we continued to maintain the project and accepted new contributions. The preprint was updated in January 2018, and the article was accepted by the journal in March 2018 [3]. As of March 06, 2019, the Deep Review repository accumulated 755 git commits, 317 merged pull requests, 609 issues, and 819 GitHub stars. The notebook to generate this figure can be interactively launched using Binder [10], enabling users to explore alternative visualizations or analyses of the source data." id="fig:contrib" style="width:100.0%" /><figcaption><span>Figure 2:</span> <strong>Deep Review contributions by author over time.</strong> The total words added to the Deep Review by each author is plotted over time (final values in parentheses). These statistics were extracted from git commit diffs of the manuscript’s Markdown source. This figure reveals the composition of written contributions to the manuscript at every point in its history. The Deep Review was initiated in August 2016, and the first complete manuscript was released as a preprint <span class="citation" data-cites="tJKvnIaZ">[<a href="#ref-tJKvnIaZ">9</a>]</span> in May 2017. While the article was under review, we continued to maintain the project and accepted new contributions. The preprint was updated in January 2018, and the article was accepted by the journal in March 2018 <span class="citation" data-cites="PZMP42Ak">[<a href="#ref-PZMP42Ak">3</a>]</span>. As of March 06, 2019, the Deep Review repository accumulated 755 git commits, 317 merged pull requests, 609 issues, and 819 GitHub stars. The notebook to generate this figure can be <a href="https://mybinder.org/v2/gh/greenelab/meta-review/binder?filepath=analyses/deep-review-contrib/02.contrib-viz.ipynb">interactively launched using Binder</a> <span class="citation" data-cites="Q20Bxdsr">[<a href="#ref-Q20Bxdsr">10</a>]</span>, enabling users to explore alternative visualizations or analyses of the source data.</figcaption>
</figure>
<h2 id="manubot">Manubot</h2>
<p>Manubot is a system for writing scholarly manuscripts via GitHub that is built upon our <a href="https://github.com/manubot/manubot">Python package</a> of the same name. With Manubot, manuscripts are written as plain-text Markdown files, which is well suited for version control using git. The Markdown standard itself provides limited yet crucial formatting syntax, including the ability to embed images and format text via bold, italics, hyperlinks, headers, inline code, codeblocks, blockquotes, and numbered or bulleted lists. In addition, Manubot relies on extensions from <a href="https://pandoc.org/MANUAL.html#pandocs-markdown">Pandoc Markdown</a> to enable citations, tables, captions, and equations specified using the popular TeX math syntax.</p>
<p>Manubot includes an additional layer of citation processing, currently unique to the system. All citations point to a standard identifier, for which Manubot automatically retrieves bibliographic metadata. Table <a href="#tbl:citations">2</a> presents the supported identifiers and example citations before and after Manubot processing. Authors can optionally define citation tags to provide short readable alternatives to the citation identifiers. Metadata is exported to <a href="http://citeproc-js.readthedocs.io/en/latest/csl-json/markup.html#items">Citation Style Language (CSL) JSON Data Items</a>, an open standard that is widely supported by reference managers <span class="citation" data-cites="9KfVIq3s K7WVgf8X">[<a href="#ref-9KfVIq3s">11</a>,<a href="#ref-K7WVgf8X">12</a>]</span>. However, sometimes external resources provide Manubot with invalid CSL Data, which can cause errors with downstream citation processors, such as <a href="http://hackage.haskell.org/package/pandoc-citeproc">pandoc-citeproc</a>. Therefore, Manubot removes invalid fields according to the <a href="https://github.com/citation-style-language/schema">CSL Data specification</a>. In cases where automatic retrieval of metadata fails or produces incorrect references — which is most common for URL citations — users can manually provide the correct CSL JSON. Manual CSL JSON also supports references without standard identifiers, such as print-only newspaper articles.</p>
<a name="tbl:citations"></a>
<table>
<caption><span>Table 2:</span> <strong>Citation types supported by Manubot.</strong> Manubot allows users to cite different types of persistent identifiers, as shown in this table. Metadata source indicates the primary resource used to retrieve bibliographic metadata. For certain identifier types, additional metadata sources are queried should the primary fail. For example, when translation-server ISBN lookup fails, Manubot tries Wikipedia’s <a href="https://www.mediawiki.org/wiki/Citoid">Citoid</a> service followed by the <a href="https://github.com/xlcnd/isbnlib">isbnlib</a> Python package. When translation-server URL lookup fails, Manubot then tries <a href="http://greycite.knowledgeblog.org/">Greycite</a> <span class="citation" data-cites="GKPtRdAw">[<a href="#ref-GKPtRdAw">13</a>]</span>. Raw citations enable citing works when no supported persistent identifiers exist, but require that the user specifies the metadata. Finally, authors may optionally map a named tag to any of the supported identifier types. In this example, the tag <code>avasthi-preprints</code> represents the DOI identifier <code>doi:10.7554/eLife.38532</code>. </caption>
<colgroup>
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
</colgroup>
<thead>
<tr class="header">
<th>Identifier</th>
<th>Metadata source</th>
<th>Example citation</th>
<th>Processed citation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Digital Object Identifier (DOI)</td>
<td>DOI <a href="https://citation.crosscite.org/docs.html">Content Negotiation</a></td>
<td><code>doi:10.1098/rsif.2017.0387</code></td>
<td><span class="citation" data-cites="PZMP42Ak">[<a href="#ref-PZMP42Ak">3</a>]</span></td>
</tr>
<tr class="even">
<td>shortDOI</td>
<td>DOI Proxy Server REST API</td>
<td><code>doi:10/gddkhn</code></td>
<td><span class="citation" data-cites="PZMP42Ak">[<a href="#ref-PZMP42Ak">3</a>]</span></td>
</tr>
<tr class="odd">
<td>PubMed Identifier (PMID)</td>
<td>NCBI <a href="https://www.ncbi.nlm.nih.gov/books/NBK25501/">E-utilities</a></td>
<td><code>pmid:25851694</code></td>
<td><span class="citation" data-cites="LfJGtB83">[<a href="#ref-LfJGtB83">14</a>]</span></td>
</tr>
<tr class="even">
<td>PubMed Central Identifier (PMCID)</td>
<td>NCBI <a href="https://api.ncbi.nlm.nih.gov/lit/ctxp/">Literature Citation Exporter</a></td>
<td><code>pmcid:PMC4719068</code></td>
<td><span class="citation" data-cites="12sHvZy1a">[<a href="#ref-12sHvZy1a">2</a>]</span></td>
</tr>
<tr class="odd">
<td>arXiv ID</td>
<td><a href="https://arxiv.org/help/api/index">arXiv API</a></td>
<td><code>arxiv:1502.04015v1</code></td>
<td><span class="citation" data-cites="Y2XyzLMc">[<a href="#ref-Y2XyzLMc">15</a>]</span></td>
</tr>
<tr class="even">
<td>International Standard Book Number (ISBN)</td>
<td><a href="https://github.com/zotero/translation-server">Zotero translation-server</a></td>
<td><code>isbn:9780262517638</code></td>
<td><span class="citation" data-cites="zBPP9YKu">[<a href="#ref-zBPP9YKu">16</a>]</span></td>
</tr>
<tr class="odd">
<td>Web address (URL)</td>
<td>Zotero translation-server</td>
<td><code>url:https://lgatto.github.io/open-and-open/</code></td>
<td><span class="citation" data-cites="zBl3qgGT">[<a href="#ref-zBl3qgGT">17</a>]</span></td>
</tr>
<tr class="even">
<td>Wikidata ID</td>
<td>Zotero translation-server</td>
<td><code>wikidata:Q56458321</code></td>
<td><span class="citation" data-cites="QhC8yJ7V">[<a href="#ref-QhC8yJ7V">18</a>]</span></td>
</tr>
<tr class="odd">
<td>Raw</td>
<td>Provided by user</td>
<td><code>raw:dongbo-conversation</code></td>
<td><span class="citation" data-cites="l625msOK">[<a href="#ref-l625msOK">19</a>]</span></td>
</tr>
<tr class="even">
<td>Tag</td>
<td>Source for tagged identifier</td>
<td><code>tag:avasthi-preprints</code></td>
<td><span class="citation" data-cites="pqBLIXzp">[<a href="#ref-pqBLIXzp">20</a>]</span></td>
</tr>
</tbody>
</table>
<p>Manubot formats bibliographies according to a <a href="http://citationstyles.org/">CSL</a> style specification. Styles define how references are constructed from bibliographic metadata, controlling layout details such as the maximum number of authors to list per reference. Manubot’s default style emphasizes titles and electronic (rather than print) identifiers and applies numeric-style citations <span class="citation" data-cites="aAKJEII">[<a href="#ref-aAKJEII">21</a>]</span>. Alternatively, users can also choose from thousands of <a href="http://editor.citationstyles.org/searchByName/">predefined styles</a> or build their own <span class="citation" data-cites="w4n6Qtun">[<a href="#ref-w4n6Qtun">22</a>]</span>. As a result, adopting the specific bibliographic format required by a journal usually just requires specifying the style’s source URL in the Manubot configuration.</p>
<p>Manubot uses <a href="https://pandoc.org/">Pandoc</a> to convert manuscripts from Markdown to HTML, PDF, and optionally DOCX outputs. Pandoc supports conversion between additional formats — such as LaTeX, AsciiDoc, EPUB, and JATS — offering Manubot users broad interoperability. <a href="https://jats.nlm.nih.gov/">Journal Article Tag Suite</a> (JATS) is a standard XML format for scholarly articles that is used by publishers, archives, and text miners <span class="citation" data-cites="JU3KpeyB AAwqxolU bCyfIm6z">[<a href="#ref-JU3KpeyB">23</a>,<a href="#ref-AAwqxolU">24</a>,<a href="#ref-bCyfIm6z">25</a>]</span>. Pandoc’s JATS support provides an avenue to integrate Manubot with the larger JATS ecosystem. For now, the primary Manubot output is HTML intended to be viewed in a web browser.</p>
<p>Manubot performs continuous publication: every update to a manuscript’s source is automatically reflected in the online outputs. The approach uses continuous integration (CI) <span class="citation" data-cites="18w6XKsQO Qh7xTLwz lXvpQxeN">[<a href="#ref-18w6XKsQO">26</a>,<a href="#ref-Qh7xTLwz">27</a>,<a href="#ref-lXvpQxeN">28</a>]</span>, specifically via <a href="https://travis-ci.org/">Travis CI</a>, to monitor changes. When changes occur, the CI service attempts to generate an updated manuscript. If this process is error free, the CI service timestamps the manuscript and uploads the output files to the GitHub repository. Because the HTML manuscript is hosted using <a href="https://pages.github.com/">GitHub Pages</a>, the CI service automatically deploys the new manuscript version when it pushes the updated outputs to GitHub. Using CI to build the manuscript automatically catches many common errors, such as misspelled citations, invalid formatting, or misconfigured software dependencies.</p>
<p>To illustrate, the source GitHub repository for this article is <a href="https://github.com/greenelab/meta-review" class="uri">https://github.com/greenelab/meta-review</a>. When this repository changes, Travis CI <a href="https://travis-ci.org/greenelab/meta-review">rebuilds</a> the manuscript. If successful, the output is deployed back to GitHub (to dedicated <a href="https://github.com/greenelab/meta-review/tree/output"><code>output</code></a> and <a href="https://github.com/greenelab/meta-review/tree/gh-pages"><code>gh-pages</code></a> branches). As a result, <a href="https://greenelab.github.io/meta-review" class="uri">https://greenelab.github.io/meta-review</a> stays up to date with the latest HTML manuscript. Furthermore, versioned URLs, such as <a href="https://greenelab.github.io/meta-review/v/4b6396bcefd1b9c7ddf39c1d3f0b3eab2dd63f31/" class="uri">https://greenelab.github.io/meta-review/v/4b6396bcefd1b9c7ddf39c1d3f0b3eab2dd63f31/</a>, provide access to previous manuscript versions.</p>
<p>The idea of the “priority of discovery” is important to science, and Vale and Hyman discuss the importance of both disclosure and validation <span class="citation" data-cites="vHuGhm4k">[<a href="#ref-vHuGhm4k">29</a>]</span>. In their framework, disclosure occurs when a scientific output is released to the world. However, for a manuscript that is shared as it is written, being able to establish priority could be challenging. Manubot supports <a href="https://opentimestamps.org/">OpenTimestamps</a> to timestamp the HTML and PDF outputs on the Bitcoin blockchain. This procedure allows one to retrospectively prove that a manuscript version existed prior to its blockchain-verifiable timestamp <span class="citation" data-cites="Y2XyzLMc 6MR50hyY QBWMEuxW qh60RjR0 6yyYojgV">[<a href="#ref-Y2XyzLMc">15</a>,<a href="#ref-6MR50hyY">30</a>,<a href="#ref-QBWMEuxW">31</a>,<a href="#ref-qh60RjR0">32</a>,<a href="#ref-6yyYojgV">33</a>]</span>. Timestamps protect against attempts to rewrite a manuscript’s history and ensure accurate histories, potentially alleviating certain authorship or priority disputes. Because all bitcoin transactions compete for limited space on the blockchain, the fees required to send a single transaction can be high. OpenTimestamps minimizes fees by encoding many timestamps into a single Bitcoin transaction, enabling the service to be free of charge <span class="citation" data-cites="1DG704X8Q">[<a href="#ref-1DG704X8Q">34</a>]</span>. Since transactions can take up to a few days to be made, Manubot initially stores incomplete timestamps and upgrades them in future continuous deployment builds. We find that this asynchronous design with timestamps precise to the day is suitable for the purposes of scientific writing.</p>
<p>Manubot and its dependencies are free of charge and largely open source. It does rely on gratis services from two proprietary platforms: GitHub and Travis CI. Fortunately, lock-in to these services is minimal, and several substitutes already exist. Manubot provides a substantial step towards end-to-end document reproducibility, where every figure or piece of data in a manuscript can be traced back to its origin <span class="citation" data-cites="sWD9uVuF">[<a href="#ref-sWD9uVuF">35</a>]</span> and is well suited for preserving provenance. For example, figures can be specified using versioned URLs that refer to the code that created them. In addition, manuscripts can be templated, so that numerical values or tables are inserted directly from the repository that created them. An <a href="https://github.com/manubot/rootstock">example repository</a> demonstrates Manubot’s features and serves as a template for users to write their own manuscript with Manubot.</p>
<p>Since its creation to facilitate the Deep Review, Manubot has been used to write a variety of scholarly documents. The <a href="https://github.com/greenelab/scihub-manuscript">Sci-Hub Coverage Study</a> — performed openly on GitHub from its inception — investigated Sci-Hub’s repository of pirated articles <span class="citation" data-cites="IhliSZDo">[<a href="#ref-IhliSZDo">36</a>]</span>. Sci-Hub <a href="https://github.com/greenelab/scihub-manuscript/issues/17">reviewed</a> the initial preprint from this study in a series of tweets, pointing out a major error in one of the analyses. Within hours, the authors used Markdown’s strikethrough formatting in Manubot to cross-out the errant sentences (<a href="https://github.com/greenelab/scihub-manuscript/commit/8fcd0cd665f6fb5f39bed7e26b940aa27d4770ba">commit</a>, <a href="https://greenelab.github.io/scihub-manuscript/v/8fcd0cd665f6fb5f39bed7e26b940aa27d4770ba/">versioned manuscript</a>), thereby alerting readers to the mistake and preventing further propagation of misinformation. One month later, a larger <a href="https://github.com/greenelab/scihub-manuscript/pull/19">set of revisions</a> explained the error in more detail and was included in a second version of the preprint. As such, continuous publishing via Manubot helped the authors address the error without delay, while retaining a public version history of the process. This Sci-Hub Coverage Study preprint was the <a href="http://web.archive.org/web/20171221221858/http://www.prepubmed.org/top_preprints/">most viewed</a> 2017 <em>PeerJ Preprint</em>, while the Deep Review was the most viewed 2017 <em>bioRxiv</em> preprint <span class="citation" data-cites="9IrsqXRa">[<a href="#ref-9IrsqXRa">37</a>]</span>. Hence, in Manubot’s first year, two of the most popular preprints were written using its collaborative, open, and review-driven authoring process.</p>
<p>Additional research studies in progress are being authored using Manubot, spanning the fields of <a href="https://vsmalladi.github.io/tfsee-manuscript/">genomics</a>, <a href="https://openclimatedata.github.io/global-emissions/">climate science</a>, and <a href="https://yt-project.github.io/yt-3.0-paper/">data visualization</a>. Manubot is also being used for documents beyond traditional journal publications, such as <a href="https://greenelab.github.io/manufund-2018/">grant proposals</a>, <a href="https://greenelab.github.io/czi-hca-report/">progress reports</a>, <a href="https://zietzm.github.io/Vagelos2017/">undergraduate research reports</a> <span class="citation" data-cites="15nwuvjrA">[<a href="#ref-15nwuvjrA">38</a>]</span>, <a href="https://slochower.github.io/synthetic-motor-literature/">literature reviews</a>, and lab notebooks. Manuscripts written with other authoring systems have been successfully ported to Manubot, including the <a href="https://git.dhimmel.com/bitcoin-whitepaper/">Bitcoin Whitepaper</a> <span class="citation" data-cites="u9DGTIX">[<a href="#ref-u9DGTIX">39</a>]</span> and <a href="https://git.dhimmel.com/rephetio-manuscript/">Project Rephetio manuscript</a> <span class="citation" data-cites="O21tn8vf">[<a href="#ref-O21tn8vf">40</a>]</span>. Finally, the Kipoi model zoo for genomics <span class="citation" data-cites="14cVrrqP1">[<a href="#ref-14cVrrqP1">41</a>]</span> uses Manubot’s citation functionality to automatically extract model authors.</p>
<h3 id="citation-utility">Citation utility</h3>
<p>To make citation-by-identifier easily usable outside of Manubot manuscripts, we created the <code>manubot cite</code> command line utility, available as a <a href="https://pypi.org/project/manubot/">Python package</a>. This utility takes a list of citations and returns either a rendered bibliography or CSL Data Items (i.e. JSON-formatted reference metadata). For example, the following command outputs a Markdown reference list for the two specified articles according to the bibliographic style of <em>PeerJ</em>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" title="1"><span class="ex">manubot</span> cite --render --format=markdown \</a>
<a class="sourceLine" id="cb1-2" title="2">  --csl=https://github.com/citation-style-language/styles/raw/master/peerj.csl \</a>
<a class="sourceLine" id="cb1-3" title="3">  pmid:29618526 doi:10.1038/550143a</a></code></pre></div>
<h2 id="authorship">Authorship</h2>
<p>To determine authorship for the Deep Review, we followed the International Committee of Medical Journal Editors (ICMJE) <a href="http://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html">guidelines</a> and used GitHub to track contributions. ICMJE recommends authors substantially contribute to, draft, approve, and agree to be accountable for the manuscript. We acknowledged other contributors who did not meet all four criteria, including contributors who provided text but did not review and approve the complete manuscript. Although these criteria provided a straightforward, equitable way to determine who would be an author, they did not produce a traditionally ordered author list. In biomedical journals, the convention is that the first and last authors made the most substantial contributions to the manuscript. This convention can be difficult to reconcile in a collaborative effort. Using git, we could quantify the number of commits each author made or the number of sentences an author wrote or edited, but these metrics discount intellectual contributions such as discussing primary literature and reviewing pull requests. However, there is no objective system to compare and weight the different types of contributions and produce an ordered author list.</p>
<p>To address this issue, we generalized the concept of “co-first” authorship, in which two or more authors are denoted as making equal contributions to a paper. We defined four types of contributions <span class="citation" data-cites="PZMP42Ak">[<a href="#ref-PZMP42Ak">3</a>]</span>, from major to minor, and reviewed the GitHub discussions and commits to assign authors to these categories. A randomized algorithm then arbitrarily ordered authors within each contribution category, and we combined the category-specific author lists to produce a traditional ordering. The randomization procedure was shared with the authors in advance (pre-registered) and run in a deterministic manner. Given the same author contributions, it always produced the same ordered author list. We annotated the author list to indicate that author order was partly randomized and emphasize that the order did not indicate one author contributed more than another from the same category. The Deep Review author ordering procedure is not inherent to writing with Manubot but illustrates the authorship possibilities when all contributions are publicly tracked and recorded.</p>
<h2 id="discussion">Discussion</h2>
<h3 id="collaborative-review-manuscripts">Collaborative review manuscripts</h3>
<p>The open scholarly writing Manubot enables has particular benefits for review articles, which present the state of the art in a scientific field <span class="citation" data-cites="Rhm4AK0j">[<a href="#ref-Rhm4AK0j">42</a>]</span>. Literature reviews are typically written in private by an invited team of colleagues. In contrast, broadly opening the process to anyone engaged in the topic — such that planning, organizing, writing, and editing occur collaboratively in a public forum where anyone is welcome to participate — can maximize a review’s value. Open drafting of reviews is especially helpful for capturing state-of-the-art knowledge about rapidly advancing research topics at the intersection of existing disciplines where contributors bring diverse opinions and expertise.</p>
<p>Writing review articles in a public forum allows review authors to engage with the original researchers to clarify their methods and results and present them accurately, as exemplified <a href="https://github.com/greenelab/deep-review/issues/213">here</a>. Additionally, discussing manuscripts in the open generates valuable pre-publication peer review of preprints <span class="citation" data-cites="pqBLIXzp">[<a href="#ref-pqBLIXzp">20</a>]</span> or post-publication peer review <span class="citation" data-cites="LfJGtB83 jYs2OUFW HflJ6Hy5">[<a href="#ref-LfJGtB83">14</a>,<a href="#ref-jYs2OUFW">43</a>,<a href="#ref-HflJ6Hy5">44</a>]</span>. Because incentives to provide public peer review of existing literature <span class="citation" data-cites="uw5bep8P">[<a href="#ref-uw5bep8P">45</a>]</span> are lacking, open collaborative reviews — where authorship is open to anyone who makes a valid contribution — could help spur more post-publication peer review.</p>
<h3 id="additional-collaborative-writing-projects">Additional collaborative writing projects</h3>
<p>The Deep Review was not the first scholarly manuscript written online via an open collaborative process. In 2013, two dozen mathematicians created the 600-page Homotopy Type Theory book, writing collaboratively in LaTeX on GitHub <span class="citation" data-cites="RExXs8is MhC1nPDK">[<a href="#ref-RExXs8is">46</a>,<a href="#ref-MhC1nPDK">47</a>]</span>. Two technical books on cryptocurrency — <a href="https://github.com/bitcoinbook/bitcoinbook">Mastering Bitcoin</a> and <a href="https://github.com/ethereumbook/ethereumbook">Mastering Ethereum</a> — written on GitHub in AsciiDoc format have engaged hundreds of contributors. Both Homotopy Type Theory and Mastering Bitcoin continue to be maintained years after their initial publication. A 2017 perspective on the future of peer review was written collaboratively on Overleaf, with contributions from 32 authors <span class="citation" data-cites="52Q1v5nS">[<a href="#ref-52Q1v5nS">48</a>]</span>. While debate was raging over tightening the default threshold for statistical significance, nearly 150 scientists contributed to a Google Doc discussion that was condensed into a traditional journal commentary <span class="citation" data-cites="82ZjWq3i HXpnCQu1">[<a href="#ref-82ZjWq3i">49</a>,<a href="#ref-HXpnCQu1">50</a>]</span>. The greatest success to date of open collaborative writing is arguably Wikipedia, whose English version contains over 5.5 million articles. Wikipedia scaled encyclopedias <a href="https://en.wikipedia.org/wiki/Wikipedia:Size_comparisons">far beyond</a> any privately-written alternative. These examples illustrate how open collaborative writing can scale scholarly manuscripts where diverse opinion and expertise are paramount beyond what would otherwise be possible.</p>
<p>Open writing also presents new opportunities for distributing scholarly communication. Though it is still valuable to have versioned drafts of a manuscript with digital identifiers, journal publication may not be the terminal endpoint for collaborative manuscripts. After releasing the first version of the Deep Review <span class="citation" data-cites="tJKvnIaZ">[<a href="#ref-tJKvnIaZ">9</a>]</span>, 14 new contributors updated the manuscript (Figure <a href="#fig:contrib">2</a>). Existing authors continue to discuss new literature, <a href="https://github.com/greenelab/deep-review/">creating a living document</a>. Manubot provides an ideal platform for perpetual reviews <span class="citation" data-cites="Xs2yPQcr H0XkaC8S">[<a href="#ref-Xs2yPQcr">51</a>,<a href="#ref-H0XkaC8S">52</a>]</span>.</p>
<p>Concepts for the future of scholarly publishing extend beyond collaborative writing <span class="citation" data-cites="WDvu1SAV ILhLpgTs">[<a href="#ref-WDvu1SAV">53</a>,<a href="#ref-ILhLpgTs">54</a>]</span>. Bookdown <span class="citation" data-cites="1BISa1RLr">[<a href="#ref-1BISa1RLr">55</a>]</span> and Pandoc Scholar <span class="citation" data-cites="17wKkS4DV">[<a href="#ref-17wKkS4DV">56</a>]</span> both extend traditional Markdown to better support publishing. Examples of continuous integration to automate manuscript generation include <a href="https://github.com/ewanmellor/gh-publisher">gh-publisher</a> and Continuous Publishing <span class="citation" data-cites="nqeDrtsc">[<a href="#ref-nqeDrtsc">57</a>]</span>, which was used to produce the book Opening Science <span class="citation" data-cites="ujrfOBM1">[<a href="#ref-ujrfOBM1">58</a>]</span>. Binder <span class="citation" data-cites="Q20Bxdsr">[<a href="#ref-Q20Bxdsr">10</a>]</span>, Distill journal articles <span class="citation" data-cites="MHNCSD5I">[<a href="#ref-MHNCSD5I">59</a>]</span>, Idyll <span class="citation" data-cites="1ESo5MNnB">[<a href="#ref-1ESo5MNnB">60</a>]</span>, and Stencila <span class="citation" data-cites="BWMf57EM">[<a href="#ref-BWMf57EM">61</a>]</span> support manuscripts with interactive graphics and close integration with the underlying code. As an open source project, Manubot can be extended to adopt best practices from these other emerging platforms.</p>
<p>Several other open science efforts are GitHub-based like our collaborative writing process. ReScience <span class="citation" data-cites="8o1nWux7">[<a href="#ref-8o1nWux7">62</a>]</span>, the Journal of Open Source Software <span class="citation" data-cites="xpw2aizK">[<a href="#ref-xpw2aizK">63</a>]</span>, and some other <a href="http://www.theoj.org/">Open Journals</a> rely on GitHub for peer review and hosting. Distill uses GitHub for transparent peer review and post-publication peer review <span class="citation" data-cites="1ESYVbN4H">[<a href="#ref-1ESYVbN4H">64</a>]</span>. GitHub is increasingly used for resource curation <span class="citation" data-cites="vf9t7xMG">[<a href="#ref-vf9t7xMG">65</a>]</span>, and collaborative scholarly reviews combine literature curation with discussion and interpretation.</p>
<h3 id="limitations">Limitations</h3>
<p>There are potential limitations of our GitHub-based approach. Because our review manuscript pertained to a computational topic, most of the authors had computational backgrounds, including previous experience with version control workflows and GitHub. In other disciplines, collaborative writing via GitHub and Manubot could present a steeper barrier to entry and deter participants. In addition, git carefully tracks all revisions to the manuscript text but not the surrounding conversations that take place through GitHub issues and pull requests. These discussions must be archived to ensure that important decisions about the manuscript are preserved and authors receive credit for intellectual contributions that are not directly reflected in the manuscript’s text. GitHub supports programmatic access to issues, pull requests, and reviews so tracking these conversations is feasible in the future.</p>
<p>In the Deep Review, we established <a href="https://github.com/greenelab/deep-review/blob/v0.9-preprint/CONTRIBUTING.md">contributor guidelines</a> that discussed norms in the areas of text contribution, peer review, and authorship, which we identified in advance as potential areas of disagreement. Our contributor guidelines required verifiable participation: either directly attributable changes to the text or participation in the discussion on GitHub. These guidelines did not discuss broader community norms that may have improved inclusiveness. It is also important to consider how the move to an open contribution model affects under-represented minority members of the scientific community <span class="citation" data-cites="zBl3qgGT">[<a href="#ref-zBl3qgGT">17</a>]</span>. Recent work has identified clear social norms and processes as helpful to maintaining a collaborative culture <span class="citation" data-cites="NuDPNceu">[<a href="#ref-NuDPNceu">66</a>]</span>. <!-- cited a previous version of doi:10.1080/08874417.2009.11645338 (see https://git.io/vdSSR) --> Conferences and open source projects have used codes of conduct to establish these norms <span class="citation" data-cites="HPKoE9m3 aRZRiJPk">[<a href="#ref-HPKoE9m3">67</a>,<a href="#ref-aRZRiJPk">68</a>]</span>. We would encourage the maintainers of similar projects to consider broader codes of conduct for project participants that build on social as well as academic norms.</p>
<h3 id="manubot-in-the-context-of-open-science">Manubot in the context of open science</h3>
<p>Science is undergoing a transition towards openness. The internet provides a global information commons, where scholarship can be publicly shared at a minimal cost. For example, open access publishing provides an economic model that encourages maximal dissemination and reuse of scholarly articles <span class="citation" data-cites="zBPP9YKu PuP45jrB HQfvK1OF">[<a href="#ref-zBPP9YKu">16</a>,<a href="#ref-PuP45jrB">69</a>,<a href="#ref-HQfvK1OF">70</a>]</span>. More broadly, open licensing solves legal barriers to content reuse, enabling any type of scholarly output to become part of the commons <span class="citation" data-cites="g6WVoxNy 137tbemL9">[<a href="#ref-g6WVoxNy">71</a>,<a href="#ref-137tbemL9">72</a>]</span>. The opportunity to reuse data and code for new investigations, as well as a push for increased reproducibility, has begot a movement to make all research outputs public, unless there are bonafide privacy or security concerns <span class="citation" data-cites="gvyja7v1 rgo1TZr 1A97a4UwU">[<a href="#ref-gvyja7v1">73</a>,<a href="#ref-rgo1TZr">74</a>,<a href="#ref-1A97a4UwU">75</a>]</span>. New tools and services make it increasingly feasible to publicly share the unabridged methods of a study, especially for computational research, which consists solely of software and data.</p>
<p>Greater openness in both research methods and publishing creates an opportunity to redefine peer review and the role journals play in communicating science <span class="citation" data-cites="52Q1v5nS">[<a href="#ref-52Q1v5nS">48</a>]</span>. At the extreme is real-time open science, whereby studies are performed entirely in the open from their inception <span class="citation" data-cites="17EdosXzD">[<a href="#ref-17EdosXzD">76</a>]</span>. Many such research projects have now been completed, benefiting from the associated early-stage peer review, additional opportunity for online collaboration, and increased visibility <span class="citation" data-cites="1pWYlPj4 O21tn8vf">[<a href="#ref-O21tn8vf">40</a>,<a href="#ref-1pWYlPj4">77</a>]</span>.</p>
<p>Manubot is an ideal authoring protocol for real-time open science, especially for projects that are already using an open source software workflow to manage their research. While Manubot does require technical expertise, the benefits are manyfold. Specifically, Manubot demonstrates a system for publishing that is transparent, reproducible, immediate, permissionless, versioned, automated, collaborative, open, linked, provenanced, decentralized, hackable, interactive, annotated, and free of charge. These attributes empower integrating Manubot with an ecosystem of other community-driven tools to make science as open and collaborative as possible.</p>
<h2 id="code-and-data-availability">Code and data availability</h2>
<p>The source code and data for this manuscript are available at <a href="https://github.com/greenelab/meta-review" class="uri">https://github.com/greenelab/meta-review</a> and archived via Software Heritage identifier <a href="https://archive.softwareheritage.org/swh:1:dir:5e644c3a487081b272b2c9b52bcd55caa89c4f85;origin=https://github.com/greenelab/meta-review/"><code>swh:1:dir:5e644c3a487081b272b2c9b52bcd55caa89c4f85</code></a>. Source code for Manubot resides in the following repositories:</p>
<ul>
<li><a href="https://github.com/manubot/manubot" class="uri">https://github.com/manubot/manubot</a> (<a href="https://gitlab.com/manubot/manubot">GitLab mirror</a>, archived at <a href="https://archive.softwareheritage.org/swh:1:dir:1f820c7aa29a5971e7e8e129cc62682131cda19f;origin=https://github.com/greenelab/manubot/"><code>swh:1:dir:1f820c7aa29a5971e7e8e129cc62682131cda19f</code></a>, packaged on <a href="https://pypi.org/project/manubot/">PyPI</a>)</li>
<li><a href="https://github.com/manubot/rootstock" class="uri">https://github.com/manubot/rootstock</a> (<a href="https://gitlab.com/manubot/rootstock">GitLab mirror</a>, archived at <a href="https://archive.softwareheritage.org/swh:1:dir:dfdf5d0fde7fa32d6559c817f37878de93e32916;origin=https://github.com/greenelab/manubot-rootstock/"><code>swh:1:dir:dfdf5d0fde7fa32d6559c817f37878de93e32916</code></a>).</li>
</ul>
<h2 id="acknowledgments">Acknowledgments</h2>
<p>We would like to thank the authors of the Deep Review who helped us test collaborative writing with Manubot. The authors who responded favorably to being acknowledged are Paul-Michael Agapow, Amr M. Alexandari, Brett K. Beaulieu-Jones, Anne E. Carpenter, Travers Ching, Evan M. Cofer, Dave DeCaprio, Brian T. Do, Enrico Ferrero, David J. Harris, Michael M. Hoffman, Alexandr A. Kalinin, Anshul Kundaje, Jack Lanchantin, Christopher A. Lavender, Benjamin J. Lengerich, Zhiyong Lu, Yifan Peng, Yanjun Qi, Gail L. Rosen, Avanti Shrikumar, Srinivas C. Turaga, Gregory P. Way, Laura K. Wiley, Stephen Woloszynek, Wei Xie, Jinbo Xu, and Michael Zietz. In addition, we thank Ogun Adebali, Evan M. Cofer, and Robert Gieseke for contributing to the Manubot template manuscript. We are grateful for additional Manubot discussion and testing by Alexander Dunkel, Ansel Halliburton, Achintya Rao, and other GitHub users. Setup and maintenance of the Zotero translation-server for Manubot usage was performed by Dongbo Hu. We thank John MacFarlane for assistance with pandoc and the global Binder team for advice on Binder.</p>
<h2 id="funding">Funding</h2>
<p>DSH and CSG were supported by <a href="https://sloan.org/grant-detail/8501">Grant G-2018-11163</a> from the Alfred P. Sloan Foundation and <a href="https://www.moore.org/grant-detail?grantId=GBMF4552">Grant GBMF4552</a> from the Gordon and Betty Moore Foundation. VSM was supported by <a href="http://www.cprit.state.tx.us/files/funded-grants/RP150596.pdf">Grant RP150596</a> from the Cancer Prevention and Research Institute of Texas.</p>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-16CgM2x0z">
<p>1. <strong>TechBlog: “Manubot” powers a crowdsourced “deep-learning” review</strong><br />
Jeffrey Perkel<br />
<em>Naturejobs</em> (2018-02-20) <a href="http://blogs.nature.com/naturejobs/2018/02/20/techblog-manubot-powers-a-crowdsourced-deep-learning-review/">http://blogs.nature.com/naturejobs/2018/02/20/techblog-manubot-powers-a-crowdsourced-deep-learning-review/</a></p>
</div>
<div id="ref-12sHvZy1a">
<p>2. <strong>Crowdsourcing in biomedicine: challenges and opportunities</strong><br />
Ritu Khare, Benjamin M Good, Robert Leaman, Andrew I Su, Zhiyong Lu<br />
<em>Briefings in bioinformatics</em> (2016-01) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4719068/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4719068/</a><br />
DOI: <a href="https://doi.org/10.1093/bib/bbv021">10.1093/bib/bbv021</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25888696">25888696</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4719068">PMC4719068</a></p>
</div>
<div id="ref-PZMP42Ak">
<p>3. <strong>Opportunities and obstacles for deep learning in biology and medicine</strong><br />
Travers Ching, Daniel S. Himmelstein, Brett K. Beaulieu-Jones, Alexandr A. Kalinin, Brian T. Do, Gregory P. Way, Enrico Ferrero, Paul-Michael Agapow, Michael Zietz, Michael M. Hoffman, … Casey S. Greene<br />
<em>Journal of The Royal Society Interface</em> (2018-04) <a href="https://doi.org/gddkhn">https://doi.org/gddkhn</a><br />
DOI: <a href="https://doi.org/10.1098/rsif.2017.0387">10.1098/rsif.2017.0387</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29618526">29618526</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5938574">PMC5938574</a></p>
</div>
<div id="ref-AylLD9F8">
<p>4. <strong>Scientific writing: the online cooperative</strong><br />
Jeffrey M. Perkel<br />
<em>Nature</em> (2014-10-01) <a href="https://doi.org/gbqsnd">https://doi.org/gbqsnd</a><br />
DOI: <a href="https://doi.org/10.1038/514127a">10.1038/514127a</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25279924">25279924</a></p>
</div>
<div id="ref-PlcxShQU">
<p>5. <strong>A Quick Introduction to Version Control with Git and GitHub</strong><br />
John D. Blischak, Emily R. Davenport, Greg Wilson<br />
<em>PLOS Computational Biology</em> (2016-01-19) <a href="https://doi.org/gbqsnf">https://doi.org/gbqsnf</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pcbi.1004668">10.1371/journal.pcbi.1004668</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26785377">26785377</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4718703">PMC4718703</a></p>
</div>
<div id="ref-kEX5dgzK">
<p>6. <strong>Ten Simple Rules for Taking Advantage of Git and GitHub</strong><br />
Yasset Perez-Riverol, Laurent Gatto, Rui Wang, Timo Sachsenberg, Julian Uszkoreit, Felipe da Veiga Leprevost, Christian Fufezan, Tobias Ternent, Stephen J. Eglen, Daniel S. Katz, … Juan Antonio Vizcaíno<br />
<em>PLOS Computational Biology</em> (2016-07-14) <a href="https://doi.org/gbrb39">https://doi.org/gbrb39</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pcbi.1004947">10.1371/journal.pcbi.1004947</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27415786">27415786</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4945047">PMC4945047</a></p>
</div>
<div id="ref-opQBBK06">
<p>7. <strong>Opportunities And Obstacles For Deep Learning In Biology And Medicine</strong><br />
Johnny Israeli<br />
<em>Towards Data Science</em> (2017-05-31) <a href="https://towardsdatascience.com/opportunities-and-obstacles-for-deep-learning-in-biology-and-medicine-6ec914fe18c2">https://towardsdatascience.com/opportunities-and-obstacles-for-deep-learning-in-biology-and-medicine-6ec914fe18c2</a></p>
</div>
<div id="ref-RK9sIADd">
<p>8. <strong>What Should Be Done To Tackle Ghostwriting in the Medical Literature?</strong><br />
Peter C Gøtzsche, Jerome P Kassirer, Karen L Woolley, Elizabeth Wager, Adam Jacobs, Art Gertel, Cindy Hamilton<br />
<em>PLoS Medicine</em> (2009-02-03) <a href="https://doi.org/bnzbx7">https://doi.org/bnzbx7</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pmed.1000023">10.1371/journal.pmed.1000023</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19192943">19192943</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2634793">PMC2634793</a></p>
</div>
<div id="ref-tJKvnIaZ">
<p>9. <strong>Opportunities And Obstacles For Deep Learning In Biology And Medicine</strong><br />
Travers Ching, Daniel S. Himmelstein, Brett K. Beaulieu-Jones, Alexandr A. Kalinin, Brian T. Do, Gregory P. Way, Enrico Ferrero, Paul-Michael Agapow, Michael Zietz, Michael M Hoffman, … Casey S. Greene<br />
<em>Cold Spring Harbor Laboratory</em> (2017-05-28) <a href="https://doi.org/gbpvh5">https://doi.org/gbpvh5</a><br />
DOI: <a href="https://doi.org/10.1101/142760">10.1101/142760</a></p>
</div>
<div id="ref-Q20Bxdsr">
<p>10. <strong>Binder 2.0 - Reproducible, interactive, sharable environments for science at scale</strong><br />
Project Jupyter, Matthias Bussonnier, Jessica Forde, Jeremy Freeman, Brian Granger, Tim Head, Chris Holdgraf, Kyle Kelley, Gladys Nalvarte, Andrew Osheroff, … Carol Willing<br />
<em>Proceedings of the 17th Python in Science Conference</em> (2018) <a href="https://doi.org/gfwcm6">https://doi.org/gfwcm6</a><br />
DOI: <a href="https://doi.org/10.25080/majora-4af1f417-011">10.25080/majora-4af1f417-011</a></p>
</div>
<div id="ref-9KfVIq3s">
<p>11. <strong>Reference Management</strong><br />
Martin Fenner, Kaja Scheliga, Sönke Bartling<br />
<em>Opening Science</em> (2013-12-17) <a href="https://doi.org/gbxtc8">https://doi.org/gbxtc8</a><br />
DOI: <a href="https://doi.org/10.1007/978-3-319-00026-8_8">10.1007/978-3-319-00026-8_8</a></p>
</div>
<div id="ref-K7WVgf8X">
<p>12. <strong>Comparison of Select Reference Management Tools</strong><br />
Yingting Zhang<br />
<em>Medical Reference Services Quarterly</em> (2012-01) <a href="https://doi.org/hpv">https://doi.org/hpv</a><br />
DOI: <a href="https://doi.org/10.1080/02763869.2012.641841">10.1080/02763869.2012.641841</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22289095">22289095</a></p>
</div>
<div id="ref-GKPtRdAw">
<p>13. <strong>Twenty-Five Shades of Greycite: Semantics for referencing and preservation</strong><br />
Phillip Lord, Lindsay Marshall<br />
<em>arXiv</em> (2013-04-26) <a href="https://arxiv.org/abs/1304.7151v1">https://arxiv.org/abs/1304.7151v1</a></p>
</div>
<div id="ref-LfJGtB83">
<p>14. <strong>Reviewing post-publication peer review.</strong><br />
Paul Knoepfler<br />
<em>Trends in genetics : TIG</em> (2015-04-04) <a href="https://www.ncbi.nlm.nih.gov/pubmed/25851694">https://www.ncbi.nlm.nih.gov/pubmed/25851694</a><br />
DOI: <a href="https://doi.org/10.1016/j.tig.2015.03.006">10.1016/j.tig.2015.03.006</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25851694">25851694</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4472664">PMC4472664</a></p>
</div>
<div id="ref-Y2XyzLMc">
<p>15. <strong>Decentralized Trusted Timestamping using the Crypto Currency Bitcoin</strong><br />
Bela Gipp, Norman Meuschke, André Gernandt<br />
<em>arXiv</em> (2015-02-13) <a href="https://arxiv.org/abs/1502.04015v1">https://arxiv.org/abs/1502.04015v1</a></p>
</div>
<div id="ref-zBPP9YKu">
<p>16. <strong>Open access</strong><br />
Peter Suber<br />
<em>MIT Press</em> (2012)<br />
ISBN: <a href="https://worldcat.org/isbn/9780262517638">9780262517638</a></p>
</div>
<div id="ref-zBl3qgGT">
<p>17. <strong>Open science and open science</strong><br />
Laurent Gatto<br />
(2017-06-05) <a href="https://lgatto.github.io/open-and-open/">https://lgatto.github.io/open-and-open/</a></p>
</div>
<div id="ref-QhC8yJ7V">
<p>18. <strong>Plan S: Accelerating the transition to full and immediate Open Access to scientific publications</strong><br />
cOAlition S<br />
(2018-09-04) <a href="https://www.wikidata.org/wiki/Q56458321">https://www.wikidata.org/wiki/Q56458321</a></p>
</div>
<div id="ref-l625msOK">
<p>19. <strong>Conversation with Dongbo Hu regarding how to administer a cloud server</strong><em>Greene Laboratory</em> (2018-12-19)</p>
</div>
<div id="ref-pqBLIXzp">
<p>20. <strong>Journal clubs in the time of preprints</strong><br />
Prachee Avasthi, Alice Soragni, Joshua N Bembenek<br />
<em>eLife</em> (2018-06-11) <a href="https://doi.org/gdm89h">https://doi.org/gdm89h</a><br />
DOI: <a href="https://doi.org/10.7554/elife.38532">10.7554/elife.38532</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29889024">29889024</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5995539">PMC5995539</a></p>
</div>
<div id="ref-aAKJEII">
<p>21. <strong>On author versus numeric citation styles</strong><br />
Daniel Himmelstein<br />
<em>Satoshi Village</em> (2018-03-12) <a href="https://blog.dhimmel.com/citation-styles/">https://blog.dhimmel.com/citation-styles/</a></p>
</div>
<div id="ref-w4n6Qtun">
<p>22. <strong>TechBlog: Create the perfect bibliography with the CSL Editor</strong><br />
Jeffrey Perkel<br />
<em>Naturejobs</em> (2017-05-03) <a href="http://blogs.nature.com/naturejobs/2017/05/03/techblog-create-the-perfect-bibliography-with-the-csl-editor/">http://blogs.nature.com/naturejobs/2017/05/03/techblog-create-the-perfect-bibliography-with-the-csl-editor/</a></p>
</div>
<div id="ref-JU3KpeyB">
<p>23. <strong>JATS: Journal Article Tag Suite, version 1.1</strong><br />
National Information Standards Organization<br />
(2015) <a href="http://www.niso.org/standards/z39-96-2015/">http://www.niso.org/standards/z39-96-2015/</a></p>
</div>
<div id="ref-AAwqxolU">
<p>24. <strong>Journal Article Tag Suite 1.0: National Information Standards Organization standard of journal extensible markup language</strong><br />
Sun Huh<br />
<em>Science Editing</em> (2014-08-18) <a href="https://doi.org/gbxtdk">https://doi.org/gbxtdk</a><br />
DOI: <a href="https://doi.org/10.6087/kcse.2014.1.99">10.6087/kcse.2014.1.99</a></p>
</div>
<div id="ref-bCyfIm6z">
<p>25. <strong>NISO Z39.96-201x, JATS: Journal Article Tag Suite</strong><br />
Mark H. Needleman<br />
<em>Serials Review</em> (2012-09) <a href="https://doi.org/gbxtdj">https://doi.org/gbxtdj</a><br />
DOI: <a href="https://doi.org/10.1080/00987913.2012.10765464">10.1080/00987913.2012.10765464</a></p>
</div>
<div id="ref-18w6XKsQO">
<p>26. <strong>Collaborative software development made easy</strong><br />
Andrew Silver<br />
<em>Nature</em> (2017-10-04) <a href="https://doi.org/cdvr">https://doi.org/cdvr</a><br />
DOI: <a href="https://doi.org/10.1038/550143a">10.1038/550143a</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28980652">28980652</a></p>
</div>
<div id="ref-Qh7xTLwz">
<p>27. <strong>Reproducibility of computational workflows is automated using continuous analysis</strong><br />
Brett K Beaulieu-Jones, Casey S Greene<br />
<em>Nature Biotechnology</em> (2017-03-13) <a href="https://doi.org/f9ttx6">https://doi.org/f9ttx6</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.3780">10.1038/nbt.3780</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28288103">28288103</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6103790">PMC6103790</a></p>
</div>
<div id="ref-lXvpQxeN">
<p>28. <strong>Developing a modern data workflow for evolving data</strong><br />
Glenda M Yenni, Erica M Christensen, Ellen K Bledsoe, Sarah R Supp, Renata M Diaz, Ethan P White, SK Morgan Ernest<br />
<em>Cold Spring Harbor Laboratory</em> (2018-06-12) <a href="https://doi.org/gdqbzn">https://doi.org/gdqbzn</a><br />
DOI: <a href="https://doi.org/10.1101/344804">10.1101/344804</a></p>
</div>
<div id="ref-vHuGhm4k">
<p>29. <strong>Priority of discovery in the life sciences</strong><br />
Ronald D Vale, Anthony A Hyman<br />
<em>eLife</em> (2016-06-16) <a href="https://doi.org/gcx6gx">https://doi.org/gcx6gx</a><br />
DOI: <a href="https://doi.org/10.7554/elife.16931">10.7554/elife.16931</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27310529">27310529</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4911212">PMC4911212</a></p>
</div>
<div id="ref-6MR50hyY">
<p>30. <strong>The Grey Literature — Proof of prespecified endpoints in medical research with the bitcoin blockchain</strong><a href="https://www.bgcarlisle.com/blog/2014/08/25/proof-of-prespecified-endpoints-in-medical-research-with-the-bitcoin-blockchain/">https://www.bgcarlisle.com/blog/2014/08/25/proof-of-prespecified-endpoints-in-medical-research-with-the-bitcoin-blockchain/</a></p>
</div>
<div id="ref-QBWMEuxW">
<p>31. <strong>The most interesting case of scientific irreproducibility?</strong><br />
Daniel Himmelstein<br />
<em>Satoshi Village</em> (2017-03-08) <a href="https://blog.dhimmel.com/irreproducible-timestamps/">https://blog.dhimmel.com/irreproducible-timestamps/</a></p>
</div>
<div id="ref-qh60RjR0">
<p>32. <strong>Bitcoin for the biological literature</strong><br />
Douglas Heaven<br />
<em>Nature</em> (2019-02) <a href="https://doi.org/gft5gp">https://doi.org/gft5gp</a><br />
DOI: <a href="https://doi.org/10.1038/d41586-019-00447-9">10.1038/d41586-019-00447-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30718888">30718888</a></p>
</div>
<div id="ref-6yyYojgV">
<p>33. <strong>Bitcoin: A Peer-to-Peer Electronic Cash System</strong><br />
Satoshi Nakamoto<br />
(2018-10-31) <a href="https://git.dhimmel.com/bitcoin-whitepaper/">https://git.dhimmel.com/bitcoin-whitepaper/</a></p>
</div>
<div id="ref-1DG704X8Q">
<p>34. <strong>OpenTimestamps: Scalable, Trust-Minimized, Distributed Timestamping with Bitcoin</strong><a href="https://petertodd.org/2016/opentimestamps-announcement">https://petertodd.org/2016/opentimestamps-announcement</a></p>
</div>
<div id="ref-sWD9uVuF">
<p>35. <strong>eLife supports development of open technology stack for publishing reproducible manuscripts online</strong><br />
Emily Packer<br />
(2017-09-07) <a href="https://elifesciences.org/for-the-press/e6038800/elife-supports-development-of-open-technology-stack-for-publishing-reproducible-manuscripts-online">https://elifesciences.org/for-the-press/e6038800/elife-supports-development-of-open-technology-stack-for-publishing-reproducible-manuscripts-online</a></p>
</div>
<div id="ref-IhliSZDo">
<p>36. <strong>Sci-Hub provides access to nearly all scholarly literature</strong><br />
Daniel S Himmelstein, Ariel Rodriguez Romero, Jacob G Levernier, Thomas Anthony Munro, Stephen Reid McLaughlin, Bastian Greshake Tzovaras, Casey S Greene<br />
<em>eLife</em> (2018-03-01) <a href="https://doi.org/ckcj">https://doi.org/ckcj</a><br />
DOI: <a href="https://doi.org/10.7554/elife.32822">10.7554/elife.32822</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29424689">29424689</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5832410">PMC5832410</a></p>
</div>
<div id="ref-9IrsqXRa">
<p>37. <strong>2017 in news: The science events that shaped the year</strong><br />
Ewen Callaway, Davide Castelvecchi, David Cyranoski, Elizabeth Gibney, Heidi Ledford, Jane J. Lee, Lauren Morello, Nicky Phillips, Quirin Schiermeier, Jeff Tollefson, … Alexandra Witze<br />
<em>Nature</em> (2017-12-21) <a href="https://doi.org/chnh">https://doi.org/chnh</a><br />
DOI: <a href="https://doi.org/10.1038/d41586-017-08493-x">10.1038/d41586-017-08493-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29293246">29293246</a></p>
</div>
<div id="ref-15nwuvjrA">
<p>38. <strong>Vagelos Report Summer 2017</strong><br />
Michael Zietz<br />
<em>Figshare</em> (2017-08-25) <a href="https://doi.org/gbr3pf">https://doi.org/gbr3pf</a><br />
DOI: <a href="https://doi.org/10.6084/m9.figshare77">10.6084/m9.figshare77</a></p>
</div>
<div id="ref-u9DGTIX">
<p>39. <strong>How I used the Manubot to reproduce the Bitcoin Whitepaper</strong><br />
Daniel Himmelstein<br />
<em>Steem</em> (2017-09-20) <a href="https://busy.org/@dhimmel/how-i-used-the-manubot-to-reproduce-the-bitcoin-whitepaper">https://busy.org/@dhimmel/how-i-used-the-manubot-to-reproduce-the-bitcoin-whitepaper</a></p>
</div>
<div id="ref-O21tn8vf">
<p>40. <strong>Systematic integration of biomedical knowledge prioritizes drugs for repurposing</strong><br />
Daniel Scott Himmelstein, Antoine Lizee, Christine Hessler, Leo Brueggeman, Sabrina L Chen, Dexter Hadley, Ari Green, Pouya Khankhanian, Sergio E Baranzini<br />
<em>eLife</em> (2017-09-22) <a href="https://doi.org/cdfk">https://doi.org/cdfk</a><br />
DOI: <a href="https://doi.org/10.7554/elife.26726">10.7554/elife.26726</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28936969">28936969</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5640425">PMC5640425</a></p>
</div>
<div id="ref-14cVrrqP1">
<p>41. <strong>Kipoi: accelerating the community exchange and reuse of predictive models for genomics</strong><br />
Ziga Avsec, Roman Kreuzhuber, Johnny Israeli, Nancy Xu, Jun Cheng, Avanti Shrikumar, Abhimanyu Banerjee, Daniel S Kim, Lara Urban, Anshul Kundaje, … Julien Gagneur<br />
<em>Cold Spring Harbor Laboratory</em> (2018-07-24) <a href="https://doi.org/gd24sx">https://doi.org/gd24sx</a><br />
DOI: <a href="https://doi.org/10.1101/375345">10.1101/375345</a></p>
</div>
<div id="ref-Rhm4AK0j">
<p>42. <strong>Ten Simple Rules for Writing a Literature Review</strong><br />
Marco Pautasso<br />
<em>PLoS Computational Biology</em> (2013-07-18) <a href="https://doi.org/gcx9dm">https://doi.org/gcx9dm</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pcbi.1003149">10.1371/journal.pcbi.1003149</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23874189">23874189</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3715443">PMC3715443</a></p>
</div>
<div id="ref-jYs2OUFW">
<p>43. <strong>A Stronger Post-Publication Culture Is Needed for Better Science</strong><br />
Hilda Bastian<br />
<em>PLoS Medicine</em> (2014-12-30) <a href="https://doi.org/gdm9cj">https://doi.org/gdm9cj</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pmed.1001772">10.1371/journal.pmed.1001772</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25548904">25548904</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4280106">PMC4280106</a></p>
</div>
<div id="ref-HflJ6Hy5">
<p>44. <strong>Post-Publication Peer Review: Opening Up Scientific Conversation</strong><br />
Jane Hunter<br />
<em>Frontiers in Computational Neuroscience</em> (2012) <a href="https://doi.org/gdm9cm">https://doi.org/gdm9cm</a><br />
DOI: <a href="https://doi.org/10.3389/fncom.2012.00063">10.3389/fncom.2012.00063</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22969719">22969719</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3431010">PMC3431010</a></p>
</div>
<div id="ref-uw5bep8P">
<p>45. <strong>Post-publication peer review, in all its guises, is here to stay</strong><br />
Michael Markie<br />
<em>Insights the UKSG journal</em> (2015-07-07) <a href="https://doi.org/gdm9ck">https://doi.org/gdm9ck</a><br />
DOI: <a href="https://doi.org/10.1629/uksg.245">10.1629/uksg.245</a></p>
</div>
<div id="ref-RExXs8is">
<p>46. <strong>Homotopy Type Theory: Univalent Foundations of Mathematics</strong><br />
The Univalent Foundations Program<br />
<em>Institute for Advanced Study</em> (2013) <a href="https://homotopytypetheory.org/book/">https://homotopytypetheory.org/book/</a></p>
</div>
<div id="ref-MhC1nPDK">
<p>47. <strong>The HoTT book</strong><br />
Andrej Bauer<br />
<em>Mathematics and Computation</em> (2013-06-20) <a href="http://math.andrej.com/2013/06/20/the-hott-book/">http://math.andrej.com/2013/06/20/the-hott-book/</a></p>
</div>
<div id="ref-52Q1v5nS">
<p>48. <strong>A multi-disciplinary perspective on emergent and future innovations in peer review</strong><br />
Jonathan P. Tennant, Jonathan M. Dugan, Daniel Graziotin, Damien C. Jacques, François Waldner, Daniel Mietchen, Yehia Elkhatib, Lauren B. Collister, Christina K. Pikas, Tom Crick, … Julien Colomb<br />
<em>F1000Research</em> (2017-11-01) <a href="https://doi.org/gc5tcv">https://doi.org/gc5tcv</a><br />
DOI: <a href="https://doi.org/10.12688/f1000research.12037.2">10.12688/f1000research.12037.2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29188015">29188015</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5686505">PMC5686505</a></p>
</div>
<div id="ref-82ZjWq3i">
<p>49. <strong>Nearly 100 scientists spent 2 months on Google Docs to redefine the p-value. Here’s what they came up with</strong><br />
Jop Vrieze<br />
<em>Science</em> (2018-01-18) <a href="https://doi.org/gc5tct">https://doi.org/gc5tct</a><br />
DOI: <a href="https://doi.org/10.1126/science.aat0471">10.1126/science.aat0471</a></p>
</div>
<div id="ref-HXpnCQu1">
<p>50. <strong>Justify your alpha</strong><br />
Daniel Lakens, Federico G. Adolfi, Casper J. Albers, Farid Anvari, Matthew A. J. Apps, Shlomo E. Argamon, Thom Baguley, Raymond B. Becker, Stephen D. Benning, Daniel E. Bradford, … Rolf A. Zwaan<br />
<em>Nature Human Behaviour</em> (2018-02-26) <a href="https://doi.org/gcz8f3">https://doi.org/gcz8f3</a><br />
DOI: <a href="https://doi.org/10.1038/s41562-018-0311-x">10.1038/s41562-018-0311-x</a></p>
</div>
<div id="ref-Xs2yPQcr">
<p>51. <strong>A proposal for regularly updated review/survey articles: “Perpetual Reviews”</strong><br />
David L. Mobley, Daniel M. Zuckerman<br />
<em>arXiv</em> (2015-02-03) <a href="https://arxiv.org/abs/1502.01329v2">https://arxiv.org/abs/1502.01329v2</a></p>
</div>
<div id="ref-H0XkaC8S">
<p>52. <strong>Why we need the Living Journal of Computational Molecular Science</strong><br />
Daniel M. Zuckerman, Michael R. Shirts, David L. Mobley<br />
<em>Living Journal of Computational Molecular Science</em> (2017-08-22) <a href="http://www.livecomsjournal.org/article/2031-why-we-need-the-living-journal-of-computational-molecular-science">http://www.livecomsjournal.org/article/2031-why-we-need-the-living-journal-of-computational-molecular-science</a><br />
DOI: <a href="https://doi.org/10.33011/livecoms.1.1.2031">10.33011/livecoms.1.1.2031</a></p>
</div>
<div id="ref-WDvu1SAV">
<p>53. <strong>The arXiv of the future will not look like the arXiv</strong><br />
Alberto Pepe, Matteo Cantiello, Josh Nicholson<br />
<em>Authorea</em> <a href="https://doi.org/gdqbz3">https://doi.org/gdqbz3</a><br />
DOI: <a href="https://doi.org/10.22541/au.149693987.70506124">10.22541/au.149693987.70506124</a></p>
</div>
<div id="ref-ILhLpgTs">
<p>54. <strong>TechBlog: C. Titus Brown: Predicting the paper of the future</strong><br />
C. Titus Brown<br />
<em>Naturejobs</em> (2017-06-01) <a href="http://blogs.nature.com/naturejobs/2017/06/01/techblog-c-titus-brown-predicting-the-paper-of-the-future/">http://blogs.nature.com/naturejobs/2017/06/01/techblog-c-titus-brown-predicting-the-paper-of-the-future/</a></p>
</div>
<div id="ref-1BISa1RLr">
<p>55. <strong>bookdown</strong><br />
Yihui Xie<br />
<em>Chapman &amp;Hall/CRC The R Series</em> (2016-12-21) <a href="https://doi.org/gdqbz2">https://doi.org/gdqbz2</a><br />
DOI: <a href="https://doi.org/10.1201/9781315204963">10.1201/9781315204963</a></p>
</div>
<div id="ref-17wKkS4DV">
<p>56. <strong>Formatting Open Science: agilely creating multiple document formats for academic manuscripts with Pandoc Scholar</strong><br />
Albert Krewinkel, Robert Winkler<br />
<em>PeerJ Computer Science</em> (2017-05-08) <a href="https://doi.org/gbrb4c">https://doi.org/gbrb4c</a><br />
DOI: <a href="https://doi.org/10.7717/peerj-cs.112">10.7717/peerj-cs.112</a></p>
</div>
<div id="ref-nqeDrtsc">
<p>57. <strong>Continuous Publishing</strong><br />
Martin Fenner<br />
<em>Gobbledygook</em> (2014-03-10) <a href="http://blog.martinfenner.org/2014/03/10/continuous-publishing/">http://blog.martinfenner.org/2014/03/10/continuous-publishing/</a></p>
</div>
<div id="ref-ujrfOBM1">
<p>58. <strong>Opening Science</strong><em>Springer International Publishing</em> (2014) <a href="https://doi.org/gdqbzz">https://doi.org/gdqbzz</a><br />
DOI: <a href="https://doi.org/10.1007/978-3-319-00026-8">10.1007/978-3-319-00026-8</a></p>
</div>
<div id="ref-MHNCSD5I">
<p>59. <strong>The Building Blocks of Interpretability</strong><br />
Chris Olah, Arvind Satyanarayan, Ian Johnson, Shan Carter, Ludwig Schubert, Katherine Ye, Alexander Mordvintsev<br />
<em>Distill</em> (2018-03-06) <a href="https://doi.org/gdvhz5">https://doi.org/gdvhz5</a><br />
DOI: <a href="https://doi.org/10.23915/distill.00010">10.23915/distill.00010</a></p>
</div>
<div id="ref-1ESo5MNnB">
<p>60. <strong>Announcing idyll.pub</strong><br />
Matthew Conlen, Andrew Osheroff<br />
<em>Idyll</em> (2018-06-26) <a href="https://idyll.pub/post/announcing-idyll-pub-0a3eff0661df3446a915700d/">https://idyll.pub/post/announcing-idyll-pub-0a3eff0661df3446a915700d/</a></p>
</div>
<div id="ref-BWMf57EM">
<p>61. <strong>Stencila – an office suite for reproducible research</strong><br />
Michael Aufreiter, Aleksandra Pawlik, Nokome Bentley<br />
<em>eLife Labs</em> (2018-07-02) <a href="https://elifesciences.org/labs/c496b8bb/stencila-an-office-suite-for-reproducible-research">https://elifesciences.org/labs/c496b8bb/stencila-an-office-suite-for-reproducible-research</a></p>
</div>
<div id="ref-8o1nWux7">
<p>62. <strong>Sustainable computational science: the ReScience initiative</strong><br />
Nicolas P. Rougier, Konrad Hinsen, Frédéric Alexandre, Thomas Arildsen, Lorena A. Barba, Fabien C.Y. Benureau, C. Titus Brown, Pierre de Buyl, Ozan Caglayan, Andrew P. Davison, … Tiziano Zito<br />
<em>PeerJ Computer Science</em> (2017-12-18) <a href="https://doi.org/gcx5kf">https://doi.org/gcx5kf</a><br />
DOI: <a href="https://doi.org/10.7717/peerj-cs.142">10.7717/peerj-cs.142</a></p>
</div>
<div id="ref-xpw2aizK">
<p>63. <strong>Journal of Open Source Software (JOSS): design and first-year review</strong><br />
Arfon M. Smith, Kyle E. Niemeyer, Daniel S. Katz, Lorena A. Barba, George Githinji, Melissa Gymrek, Kathryn D. Huff, Christopher R. Madan, Abigail Cabunoc Mayes, Kevin M. Moerman, … Jacob T. Vanderplas<br />
<em>PeerJ Computer Science</em> (2018-02-12) <a href="https://doi.org/gc5sjf">https://doi.org/gc5sjf</a><br />
DOI: <a href="https://doi.org/10.7717/peerj-cs.147">10.7717/peerj-cs.147</a></p>
</div>
<div id="ref-1ESYVbN4H">
<p>64. <strong>Distill Update 2018</strong><br />
Distill Editors<br />
<em>Distill</em> (2018-08-14) <a href="https://doi.org/gfbzs9">https://doi.org/gfbzs9</a><br />
DOI: <a href="https://doi.org/10.23915/distill.00013">10.23915/distill.00013</a></p>
</div>
<div id="ref-vf9t7xMG">
<p>65. <strong>The appropriation of GitHub for curation</strong><br />
Yu Wu, Na Wang, Jessica Kropczynski, John M. Carroll<br />
<em>PeerJ Computer Science</em> (2017-10-09) <a href="https://doi.org/gb3bxk">https://doi.org/gb3bxk</a><br />
DOI: <a href="https://doi.org/10.7717/peerj-cs.134">10.7717/peerj-cs.134</a></p>
</div>
<div id="ref-NuDPNceu">
<p>66. <strong>Innovating Collaborative Content Creation: The Role of Altruism and Wiki Technology</strong><br />
Christian Wagner, Pattarawan Prasarnphanich<br />
<em>2007 40th Annual Hawaii International Conference on System Sciences (HICSS’07)</em> (2007) <a href="https://doi.org/b6vqgx">https://doi.org/b6vqgx</a><br />
DOI: <a href="https://doi.org/10.1109/hicss.2007.277">10.1109/hicss.2007.277</a></p>
</div>
<div id="ref-HPKoE9m3">
<p>67. <strong>Code of conduct evaluations</strong><em>Geek Feminism Wiki</em> <a href="http://geekfeminism.wikia.com/wiki/Code_of_conduct_evaluations">http://geekfeminism.wikia.com/wiki/Code_of_conduct_evaluations</a></p>
</div>
<div id="ref-aRZRiJPk">
<p>68. <strong>Contributor Covenant: A Code of Conduct for Open Source Projects</strong><br />
Coraline Ada Ehmke<br />
(2014) <a href="https://www.contributor-covenant.org/">https://www.contributor-covenant.org/</a></p>
</div>
<div id="ref-PuP45jrB">
<p>69. <strong>The academic, economic and societal impacts of Open Access: an evidence-based review</strong><br />
Jonathan P. Tennant, François Waldner, Damien C. Jacques, Paola Masuzzo, Lauren B. Collister, Chris. H. J. Hartgerink<br />
<em>F1000Research</em> (2016-09-21) <a href="https://doi.org/gbqrbc">https://doi.org/gbqrbc</a><br />
DOI: <a href="https://doi.org/10.12688/f1000research.8460.3">10.12688/f1000research.8460.3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27158456">27158456</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4837983">PMC4837983</a></p>
</div>
<div id="ref-HQfvK1OF">
<p>70. <strong>How open science helps researchers succeed</strong><br />
Erin C McKiernan, Philip E Bourne, C Titus Brown, Stuart Buck, Amye Kenall, Jennifer Lin, Damon McDougall, Brian A Nosek, Karthik Ram, Courtney K Soderberg, … Tal Yarkoni<br />
<em>eLife</em> (2016-07-07) <a href="https://doi.org/gbqsng">https://doi.org/gbqsng</a><br />
DOI: <a href="https://doi.org/10.7554/elife.16800">10.7554/elife.16800</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27387362">27387362</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4973366">PMC4973366</a></p>
</div>
<div id="ref-g6WVoxNy">
<p>71. <strong>The Legal Framework for Reproducible Scientific Research: Licensing and Copyright</strong><br />
Victoria Stodden<br />
<em>Computing in Science &amp; Engineering</em> (2009-01) <a href="https://doi.org/b7tskf">https://doi.org/b7tskf</a><br />
DOI: <a href="https://doi.org/10.1109/mcse.2009.19">10.1109/mcse.2009.19</a></p>
</div>
<div id="ref-137tbemL9">
<p>72. <strong>Legal confusion threatens to slow data science</strong><br />
Simon Oxenham<br />
<em>Nature</em> (2016-08-03) <a href="https://doi.org/bndt">https://doi.org/bndt</a><br />
DOI: <a href="https://doi.org/10.1038/536016a">10.1038/536016a</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27488781">27488781</a></p>
</div>
<div id="ref-gvyja7v1">
<p>73. <strong>Enhancing reproducibility for computational methods</strong><br />
V. Stodden, M. McNutt, D. H. Bailey, E. Deelman, Y. Gil, B. Hanson, M. A. Heroux, J. P. A. Ioannidis, M. Taufer<br />
<em>Science</em> (2016-12-08) <a href="https://doi.org/gbr42b">https://doi.org/gbr42b</a><br />
DOI: <a href="https://doi.org/10.1126/science.aah6168">10.1126/science.aah6168</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27940837">27940837</a></p>
</div>
<div id="ref-rgo1TZr">
<p>74. <strong>The case for open computer programs</strong><br />
Darrel C. Ince, Leslie Hatton, John Graham-Cumming<br />
<em>Nature</em> (2012-02-22) <a href="https://doi.org/hqg">https://doi.org/hqg</a><br />
DOI: <a href="https://doi.org/10.1038/nature10836">10.1038/nature10836</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22358837">22358837</a></p>
</div>
<div id="ref-1A97a4UwU">
<p>75. <strong>The Open Knowledge Foundation: Open Data Means Better Science</strong><br />
Jennifer C. Molloy<br />
<em>PLoS Biology</em> (2011-12-06) <a href="https://doi.org/g3b">https://doi.org/g3b</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pbio.1001195">10.1371/journal.pbio.1001195</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22162946">22162946</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232214">PMC3232214</a></p>
</div>
<div id="ref-17EdosXzD">
<p>76. <strong>This revolution will be digitized: online tools for radical collaboration</strong><br />
C. Patil, V. Siegel<br />
<em>Disease Models &amp; Mechanisms</em> (2009-04-30) <a href="https://doi.org/fvjhcj">https://doi.org/fvjhcj</a><br />
DOI: <a href="https://doi.org/10.1242/dmm.003285">10.1242/dmm.003285</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19407323">19407323</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2675795">PMC2675795</a></p>
</div>
<div id="ref-1pWYlPj4">
<p>77. <strong>Publishing the research process</strong><br />
Daniel Mietchen, Ross Mounce, Lyubomir Penev<br />
<em>Research Ideas and Outcomes</em> (2015-12-17) <a href="https://doi.org/f3mn7d">https://doi.org/f3mn7d</a><br />
DOI: <a href="https://doi.org/10.3897/rio.1.e7547">10.3897/rio.1.e7547</a></p>
</div>
</div>
<!-- default theme -->
<style>

    /* import google fonts */
    @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
    @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

    /* -------------------------------------------------- */
    /* global */
    /* -------------------------------------------------- */

    /* all elements */
    * {
        /* force sans-serif font unless specified otherwise */
        font-family: "Open Sans", "Helvetica", sans-serif;
    }

    /* "page" element */
    body {
        position: relative;
        box-sizing: border-box;
        /* make resemble paper */
        max-width: 8.5in;
        font-size: 16px;
        line-height: 1.5;
        margin: 20px auto;
        padding: 40px;
        border-radius: 4px;
        background: #ffffff;
        box-shadow: 0 0 15px #bbbbbb;
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* "page" element */
        body {
            /* reduce padding to fit more text on screen */
            padding: 20px;
            /* remove the "paper" look */
            margin: 0;
            box-shadow: none;
        }
    }

    /* -------------------------------------------------- */
    /* headings */
    /* -------------------------------------------------- */

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        margin: 20px 0;
        padding: 0;
        font-weight: bold;
    }

    /* biggest heading */
    h1 {
        margin-top: 40px;
        margin-bottom: 30px;
        text-align: center;
    }

    /* second biggest heading */
    h2 {
        margin-top: 30px;
        padding-bottom: 5px;
        /* underline */
        border-bottom: solid 1px #bbbbbb;
    }

    /* -------------------------------------------------- */
    /* manuscript header */
    /* -------------------------------------------------- */

    /* manuscript title */
    header > h1 {
        margin: 0;
    }

    /* manuscript title caption text (ie "automatically generated on") */
    header + p {
        text-align: center;
        margin-top: 0;
    }

    /* -------------------------------------------------- */
    /* text elements */
    /* -------------------------------------------------- */

    /* links */
    a {
        color: #2196f3;
    }

    /* superscripts and subscripts */
    sub,
    sup {
        /* prevent from affecting line height */
        line-height: 0;
    }

    /* unordered and ordered lists*/
    ul,
    ol {
        /* align similarly to other elements */
        padding-left: 20px;
    }

    /* -------------------------------------------------- */
    /* section elements */
    /* -------------------------------------------------- */

    /* horizontal divider line */
    hr {
        border: none;
        height: 1px;
        background: #bbbbbb;
    }

    /* paragraphs, horizontal dividers, figures, and tables */
    p,
    hr,
    figure,
    table {
        /* treat all as "paragraphs", with consistent vertical margins */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* figures and tables */
    figure,
    table {
        /* horizontally center */
        margin-left: auto;
        margin-right: auto;
    }

    /* -------------------------------------------------- */
    /* figures */
    /* -------------------------------------------------- */

    /* figure container */
    figure {
        max-width: 100%;
        /* horizontally center */
        margin-left: auto;
        margin-right: auto;
    }

    /* figure caption */
    figcaption {
        margin: 10px 20px 0 20px;
    }

    /* figure auto-number */
    figcaption > span:first-child {
        font-weight: 600;
        margin-right: 5px;
    }

    /* figure image element */
    figure img {
        /* squash image if too wide for page */
        max-width: 100%;
        /* horizontally center */
        display: block;
        margin-left: auto;
        margin-right: auto;
    }

    /* -------------------------------------------------- */
    /* tables */
    /* -------------------------------------------------- */

    /* table container */
    table {
        border-collapse: collapse;
        border-spacing: 0;
        width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* table caption */
    caption {
        text-align: left;
        margin: 0 20px 10px 20px;
    }

    /* table auto-number */
    caption > span:first-child {
        font-weight: 600;
        margin-right: 5px;
    }

    /* table cells */
    th,
    td {
        border: solid 1px #bbbbbb;
        padding: 10px;
        /* squash table if too wide for page by forcing line breaks */
        /* put line break break anywhere in text */
        word-break: break-all;
        /* put line break between words. if not supported by browser, will fall back to break-all */
        word-break: break-word;
    }

    /* header row and even rows */
    th,
    tr:nth-child(2n) {
        background-color: #f8f8f8;
    }

    /* odd rows */
    tr:nth-child(2n + 1) {
        background-color: #ffffff;
    }

    /* -------------------------------------------------- */
    /* code */
    /* -------------------------------------------------- */

    /* multi-line code block */
    pre {
        /* give grey box background */
        padding: 10px;
        background-color: #f0f0f0;
        border: solid 1px #bbbbbb;
        border-radius: 4px;
    }

    /* inline code, ie code within normal text */
    /* a piece of code not already surrounded by a code block */
    :not(pre) > code {
        /* give grey box background */
        padding: 1px 5px;
        background-color: #f0f0f0;
        border: solid 1px #bbbbbb;
        border-radius: 4px;
    }

    /* code text */
    /* apply all children, to reach syntax highlighting sub-elements */
    code,
    code * {
        /* force monospace font */
        font-family: "Source Code Pro", "Courier New", monospace;
        /* reduce font size to match relative size of main font */
        font-size: 14px;
    }

    /* -------------------------------------------------- */
    /* quotes */
    /* -------------------------------------------------- */

    /* quoted text */
    blockquote {
        margin: 0;
        padding: 0;
        border-left: 4px solid #bbbbbb;
        /* align similarly to other elements */
        padding-left: 16px;
    }

    /* -------------------------------------------------- */
    /* mathjax */
    /* -------------------------------------------------- */

    /* all first-generation children of mathjax containers */
    .math.display > span {
        overflow: auto hidden !important;
        /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
        display: flex !important;
        justify-content: center;
        align-items: center;
        /* remove mathjax margins */
        margin: 0 !important;
    }

    /* right click menu */
    .MathJax_Menu {
        border: none !important;
        border-radius: 4px !important;
        box-shadow: 0 0 15px #bbbbbb !important;
    }

    /* -------------------------------------------------- */
    /* buttons */
    /* -------------------------------------------------- */

    /* class for styling links like buttons */
    .button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        margin: 5px;
        padding: 10px 20px;
        font-size: 13px;
        font-weight: 600;
        text-transform: uppercase;
        text-decoration: none;
        letter-spacing: 1px;
        background: none;
        color: #2196f3;
        border: solid 1px #bbbbbb;
        border-radius: 4px;
    }

    /* buttons when hovered */
    .button:hover:not([disabled]),
    .icon_button:hover:not([disabled]) {
        cursor: pointer;
        background: #f0f0f0;
    }

    /* buttons when disabled */
    .button[disabled],
    .icon_button[disabled] {
        opacity: 0.35;
        pointer-events: none;
    }

    /* class for styling buttons containg only single icon (used in plugins) */
    .icon_button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        text-decoration: none;
        margin: 0;
        padding: 0;
        background: none;
        border-radius: 4px;
        border: none;
        width: 20px;
        height: 20px;
    }

    /* -------------------------------------------------- */
    /* icons */
    /* -------------------------------------------------- */

    /* class for styling icons inline with text */
    .inline_icon {
        height: 1em;
        position: relative;
        top: 0.125em;
    }

    /* -------------------------------------------------- */
    /* when printing */
    /* -------------------------------------------------- */

    /* paper/page size and margins */
    @page {
        size: letter;
        margin-top: 0.75in;
        margin-bottom: 0.75in;
        margin-left: 0.75in;
        margin-right: 0.75in;
    }

    @media print {
        /* page element */
        body {
            /* remove margin and padding; go right up to page margins */
            margin: 0;
            padding: 0;
            /* remove "paper" effect */
            box-shadow: none;
            /* make font size smaller to fit more words on page */
            font-size: 12px;
        }

        /* all h2 headings except the first */
        h2:not(:first-of-type) {
            /* force page break */
            page-break-before: always !important;
        }

        /* always insert a page break before the element */
        .page_break_before {
            page-break-before: always !important;
        }

        /* always insert a page break after the element */
        .page_break_after {
            page-break-after: always !important;
        }

        /* avoid page break before the element */
        .page_break_before_avoid {
            page-break-before: avoid !important;
        }

        /* avoid page break after the element */
        .page_break_after_avoid {
            page-break-after: avoid !important;
        }

        /* avoid page break inside the element */
        .page_break_inside_avoid {
            page-break-inside: avoid !important;
        }
    }

    /* -------------------------------------------------- */
    /* override pandoc css quirks */
    /* -------------------------------------------------- */

    .sourceCode {
        /* prevent unsightly overflow in wide code blocks */
        overflow: auto !important;
    }

</style>
<!-- Insert Analytics Script Below -->
<script>
</script>
<!-- End Analytics Script -->
<script>
// AnchorJS minified version below.
// Source https://github.com/bryanbraun/anchorjs/blob/064abdd0987f305933ec4982af6d0c1cf2fd0814/anchor.js

/**
 * AnchorJS - v4.0.0 - 2017-06-02
 * https://github.com/bryanbraun/anchorjs
 * Copyright (c) 2017 Bryan Braun; Licensed MIT
 */
!function(A,e){"use strict";"function"==typeof define&&define.amd?define([],e):"object"==typeof module&&module.exports?module.exports=e():(A.AnchorJS=e(),A.anchors=new A.AnchorJS)}(this,function(){"use strict";function A(A){function e(A){A.icon=A.hasOwnProperty("icon")?A.icon:"",A.visible=A.hasOwnProperty("visible")?A.visible:"hover",A.placement=A.hasOwnProperty("placement")?A.placement:"right",A.class=A.hasOwnProperty("class")?A.class:"",A.truncate=A.hasOwnProperty("truncate")?Math.floor(A.truncate):64}function t(A){var e;if("string"==typeof A||A instanceof String)e=[].slice.call(document.querySelectorAll(A));else{if(!(Array.isArray(A)||A instanceof NodeList))throw new Error("The selector provided to AnchorJS was invalid.");e=[].slice.call(A)}return e}function n(){if(null===document.head.querySelector("style.anchorjs")){var A,e=document.createElement("style");e.className="anchorjs",e.appendChild(document.createTextNode("")),void 0===(A=document.head.querySelector('[rel="stylesheet"], style'))?document.head.appendChild(e):document.head.insertBefore(e,A),e.sheet.insertRule(" .anchorjs-link {   opacity: 0;   text-decoration: none;   -webkit-font-smoothing: antialiased;   -moz-osx-font-smoothing: grayscale; }",e.sheet.cssRules.length),e.sheet.insertRule(" *:hover > .anchorjs-link, .anchorjs-link:focus  {   opacity: 1; }",e.sheet.cssRules.length),e.sheet.insertRule(" [data-anchorjs-icon]::after {   content: attr(data-anchorjs-icon); }",e.sheet.cssRules.length),e.sheet.insertRule(' @font-face {   font-family: "anchorjs-icons";   src: url(data:n/a;base64,AAEAAAALAIAAAwAwT1MvMg8yG2cAAAE4AAAAYGNtYXDp3gC3AAABpAAAAExnYXNwAAAAEAAAA9wAAAAIZ2x5ZlQCcfwAAAH4AAABCGhlYWQHFvHyAAAAvAAAADZoaGVhBnACFwAAAPQAAAAkaG10eASAADEAAAGYAAAADGxvY2EACACEAAAB8AAAAAhtYXhwAAYAVwAAARgAAAAgbmFtZQGOH9cAAAMAAAAAunBvc3QAAwAAAAADvAAAACAAAQAAAAEAAHzE2p9fDzz1AAkEAAAAAADRecUWAAAAANQA6R8AAAAAAoACwAAAAAgAAgAAAAAAAAABAAADwP/AAAACgAAA/9MCrQABAAAAAAAAAAAAAAAAAAAAAwABAAAAAwBVAAIAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAMCQAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAg//0DwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAAIAAAACgAAxAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADAAAAAIAAgAAgAAACDpy//9//8AAAAg6cv//f///+EWNwADAAEAAAAAAAAAAAAAAAAACACEAAEAAAAAAAAAAAAAAAAxAAACAAQARAKAAsAAKwBUAAABIiYnJjQ3NzY2MzIWFxYUBwcGIicmNDc3NjQnJiYjIgYHBwYUFxYUBwYGIwciJicmNDc3NjIXFhQHBwYUFxYWMzI2Nzc2NCcmNDc2MhcWFAcHBgYjARQGDAUtLXoWOR8fORYtLTgKGwoKCjgaGg0gEhIgDXoaGgkJBQwHdR85Fi0tOAobCgoKOBoaDSASEiANehoaCQkKGwotLXoWOR8BMwUFLYEuehYXFxYugC44CQkKGwo4GkoaDQ0NDXoaShoKGwoFBe8XFi6ALjgJCQobCjgaShoNDQ0NehpKGgobCgoKLYEuehYXAAAADACWAAEAAAAAAAEACAAAAAEAAAAAAAIAAwAIAAEAAAAAAAMACAAAAAEAAAAAAAQACAAAAAEAAAAAAAUAAQALAAEAAAAAAAYACAAAAAMAAQQJAAEAEAAMAAMAAQQJAAIABgAcAAMAAQQJAAMAEAAMAAMAAQQJAAQAEAAMAAMAAQQJAAUAAgAiAAMAAQQJAAYAEAAMYW5jaG9yanM0MDBAAGEAbgBjAGgAbwByAGoAcwA0ADAAMABAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAH//wAP) format("truetype"); }',e.sheet.cssRules.length)}}this.options=A||{},this.elements=[],e(this.options),this.isTouchDevice=function(){return!!("ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch)},this.add=function(A){var i,o,s,c,r,a,h,l,u,d,f,g,p=[];if(e(this.options),"touch"===(g=this.options.visible)&&(g=this.isTouchDevice()?"always":"hover"),A||(A="h2, h3, h4, h5, h6"),0===(i=t(A)).length)return this;for(n(),o=document.querySelectorAll("[id]"),s=[].map.call(o,function(A){return A.id}),r=0;r<i.length;r++)if(this.hasAnchorJSLink(i[r]))p.push(r);else{if(i[r].hasAttribute("id"))c=i[r].getAttribute("id");else if(i[r].hasAttribute("data-anchor-id"))c=i[r].getAttribute("data-anchor-id");else{u=l=this.urlify(i[r].textContent),h=0;do{void 0!==a&&(u=l+"-"+h),a=s.indexOf(u),h+=1}while(-1!==a);a=void 0,s.push(u),i[r].setAttribute("id",u),c=u}d=c.replace(/-/g," "),(f=document.createElement("a")).className="anchorjs-link "+this.options.class,f.href="#"+c,f.setAttribute("aria-label","Anchor link for: "+d),f.setAttribute("data-anchorjs-icon",this.options.icon),"always"===g&&(f.style.opacity="1"),""===this.options.icon&&(f.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(f.style.lineHeight="inherit")),"left"===this.options.placement?(f.style.position="absolute",f.style.marginLeft="-1em",f.style.paddingRight="0.5em",i[r].insertBefore(f,i[r].firstChild)):(f.style.paddingLeft="0.375em",i[r].appendChild(f))}for(r=0;r<p.length;r++)i.splice(p[r]-r,1);return this.elements=this.elements.concat(i),this},this.remove=function(A){for(var e,n,i=t(A),o=0;o<i.length;o++)(n=i[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(i[o]))&&this.elements.splice(e,1),i[o].removeChild(n));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}return A});

// Enable links for selected headers
var anchors = new AnchorJS();
anchors.add("h2, h3, h4")
</script>
<!-- Enable Hypothesis annotation. https://web.hypothes.is/ -->
<script src="https://hypothes.is/embed.js" async></script>
</body>
</html>
