<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Daniel S. Himmelstein" />
  <meta name="author" content="David R. Slochower" />
  <meta name="author" content="Venkat S. Malladi" />
  <meta name="author" content="Casey S. Greene" />
  <meta name="author" content="Anthony Gitter" />
  <meta name="dcterms.date" content="2018-07-16" />
  <meta name="keywords" content="collaborative review, continuous integration, publishing, Manubot" />
  <title>Open collaborative writing with Manubot</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="github-pandoc.css">
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!-- Insert Analytics Script Below -->
  <script>
  </script>
  <!-- End Analytics Script -->
</head>
<body>
<header>
<h1 class="title">Open collaborative writing with Manubot</h1>
</header>
<p><small><em> This manuscript (<a href="https://greenelab.github.io/meta-review/v/59442cd3b5417ef45c8dd6477ec6a782495aad4b/">permalink</a>) was automatically generated from <a href="https://github.com/greenelab/meta-review/tree/59442cd3b5417ef45c8dd6477ec6a782495aad4b">greenelab/meta-review@59442cd</a> on July 16, 2018. </em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Daniel S. Himmelstein</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0002-3012-7446">0000-0002-3012-7446</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/dhimmel">dhimmel</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/dhimmel">dhimmel</a><br> <small> Department of Systems Pharmacology and Translational Therapeutics, University of Pennsylvania · Funded by GBMF4552 </small></p></li>
<li><p><strong>David R. Slochower</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0003-3928-5050">0000-0003-3928-5050</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/slochower">slochower</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/drslochower">drslochower</a><br> <small> Skaggs School of Pharmacy and Pharmaceutical Sciences, University of California, San Diego </small></p></li>
<li><p><strong>Venkat S. Malladi</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0002-0144-0564">0000-0002-0144-0564</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/vsmalladi">vsmalladi</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/katatonikkat">katatonikkat</a><br> <small> Department of Bioinformatics, University of Texas Southwestern Medical Center; Bioinformatics Core Facility, University of Texas Southwestern Medical Center · Funded by RP150596 </small></p></li>
<li><p><strong>Casey S. Greene</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0001-8713-9213">0000-0001-8713-9213</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/cgreene">cgreene</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/GreeneScientist">GreeneScientist</a><br> <small> Department of Systems Pharmacology and Translational Therapeutics, University of Pennsylvania · Funded by GBMF4552 </small></p></li>
<li><p><strong>Anthony Gitter</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0002-5324-9833">0000-0002-5324-9833</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/agitter">agitter</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/anthonygitter">anthonygitter</a><br> <small> Department of Biostatistics and Medical Informatics, University of Wisconsin-Madison; Morgridge Institute for Research · Funded by NIH U54AI117924 </small></p></li>
</ul>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<p>Scientific literature reviews are ideal for open, collaborative writing. Allowing any interested individual to contribute as an author can strengthen a review, providing broad and fair coverage of the subject matter. However, the traditional multi-author writing process breaks down at scale. We present techniques for overcoming the challenges of open manuscript writing. These include approaches for managing distributed authors and our new software, named Manubot, for automating citation and manuscript building.</p>
<h2 id="introduction" class="page_break_before">Introduction</h2>
<p>Openness in research — which includes sharing code, data, and manuscripts — benefits the researchers who practice it <span class="citation" data-cites="HQfvK1OF">[<a href="#ref-HQfvK1OF">1</a>]</span>, their scientific peers, and the public. Open scholarly writing, a form of crowdsourcing <span class="citation" data-cites="12sHvZy1a">[<a href="#ref-12sHvZy1a">2</a>]</span>, has particular benefits for review articles, which present the state of the art in a scientific field <span class="citation" data-cites="Rhm4AK0j">[<a href="#ref-Rhm4AK0j">3</a>]</span>. Literature reviews are typically written in private by a closed team of colleagues. In contrast, broadly opening the process to anyone engaged in the topic — such that planning, organizing, writing, and editing occur collaboratively in a public forum where anyone is welcome to participate — can maximize a review’s value. Open drafting of reviews is especially helpful for capturing state-of-the-art knowledge about rapidly advancing research topics at the intersection of existing disciplines where contributors bring diverse opinions and expertise.</p>
<p>Based on our experience leading a recent open review <span class="citation" data-cites="16CgM2x0z">[<a href="#ref-16CgM2x0z">4</a>]</span>, we discuss the pros and cons of open collaborative reviews as well as how they contribute to a living and frequently updated literature. Our review manuscript <span class="citation" data-cites="PZMP42Ak">[<a href="#ref-PZMP42Ak">5</a>]</span>, code-named the Deep Review, surveyed deep learning’s role in biology and precision medicine, a research area undergoing explosive growth. In addition, we introduce <a href="https://github.com/greenelab/manubot-rootstock">Manubot</a>, the infrastructure we created to enable open manuscript writing online for Deep Review, which was subsequently adopted by other projects.</p>
<p>We initiated the Deep Review by creating a GitHub repository (<a href="https://github.com/greenelab/deep-review" class="uri">https://github.com/greenelab/deep-review</a>) to coordinate and manage contributions. GitHub is a platform designed for collaborative software development that’s adaptable for collaborative writing. From the start, we made the GitHub repository public, applying a <a href="https://github.com/greenelab/deep-review/blob/master/LICENSE.md">Creative Commons Attribution License</a> to the manuscript. Next, we encouraged anyone interested to contribute by proposing changes or additions. Although we invited some specific experts to participate, most authors discovered the manuscript organically through conferences or social media, deciding to contribute without solicitation. In total, the Deep Review attracted 36 authors who were not determined in advance from 20 different institutions.</p>
<p>Writing review articles in a public forum allows review authors to engage with the original researchers to clarify their methods and results and present them accurately, as exemplified <a href="https://github.com/greenelab/deep-review/issues/213">here</a>. Additionally, discussing manuscripts in the open generates valuable post-publication peer review <span class="citation" data-cites="LfJGtB83 jYs2OUFW HflJ6Hy5">[<a href="#ref-LfJGtB83">6</a>,<a href="#ref-jYs2OUFW">7</a>,<a href="#ref-HflJ6Hy5">8</a>]</span> or pre-publication peer review of preprints <span class="citation" data-cites="pqBLIXzp">[<a href="#ref-pqBLIXzp">9</a>]</span>. Because incentives are commonly lacking to provide public peer review of existing literature <span class="citation" data-cites="uw5bep8P">[<a href="#ref-uw5bep8P">10</a>]</span>, open collaborative reviews — where authorship is open to anyone who makes a valid contribution — could help spur more post-publication peer review. However, inviting wide authorship brings many technical and social challenges such as how to fairly distribute credit, coordinate the scientific content, and collaboratively manage extensive reference lists.</p>
<p>To address these challenges, we developed a manuscript writing process using the Markdown language, the GitHub platform, and our new Manubot tool for automating manuscript generation.</p>
<h2 id="contribution-workflow">Contribution workflow</h2>
<p>There are many existing collaborative writing platforms ranging from rich text editors, which support Microsoft Word documents or similar formats, to LaTeX-based systems for technical writing <span class="citation" data-cites="AylLD9F8">[<a href="#ref-AylLD9F8">11</a>]</span> such as <a href="https://www.overleaf.com/">Overleaf</a> and <a href="https://www.authorea.com/">Authorea</a>. These platforms ideally offer version control, multiple permission levels, or other functionality to support multi-author document editing. Although they work well for editing, they lack sufficient features for managing a collaborative manuscript and attributing precise credit, which are important for open writing.</p>
<p>We adopted standard software development strategies in order to enable any contributor to edit any part of the manuscript but enforce discussion and review of all proposed changes. The GitHub platform provided support for organizing and editing the manuscript. We used GitHub <em>issues</em> for organization, opening a new issue for each paper under consideration. Within the issue, contributors summarized the research, discussed it (sometimes with the original authors), and assessed its relevance to the review. Issues also served as an open to-do list and a forum for debating the main message, themes, and topics of the review.</p>
<p>GitHub and the underlying git version control system <span class="citation" data-cites="PlcxShQU kEX5dgzK">[<a href="#ref-PlcxShQU">12</a>,<a href="#ref-kEX5dgzK">13</a>]</span> also structured the writing process. The official version of the manuscript is <em>forked</em> by individual contributors. A contributor then adds and revises files, grouping these changes into <em>commits</em>. When the changes are ready to be reviewed, the series of commits are submitted as a <em>pull request</em> through GitHub, which notifies other authors of the pending changes. GitHub’s review interface allows anyone to comment on the changes, globally or at specific lines, asking questions or requesting modifications as depicted in <span class="citation" data-cites="opQBBK06"><a href="#ref-opQBBK06">14</a></span>. Conversations during review can reference other pull requests, issues, or authors, linking the relevant people and content, as illustrated in Figure <a href="#fig:workflow">1</a>. Reviewing batches of revisions that focus on a single theme is more efficient than independently discussing isolated comments and edits, and it helps maintain consistent content and tone across different authors and reviewers. Once all requested modifications are made, the manuscript maintainers, a subset of authors with elevated GitHub permissions, formally approve the pull request and merge the changes into the official version. The process of writing and revising material can be orchestrated through GitHub with a web browser.</p>
<figure>
<img src="images/workflow.svg" alt="Figure 1: Deep Review editing workflow. Any reader can become a contributor by proposing a change through a pull request. In this example, the contributor opens an issue to discuss a manuscript modification. A maintainer and additional participant provide feedback, and the maintainer recommends creating a pull request to update the text. The contributor creates the pull request, it is reviewed by a maintainer and a participant, and the contributor updates the pull request in response. Once the pull request is approved, the maintainer merges the changes into the official version of the manuscript." id="fig:workflow" style="width:5in" /><figcaption><span>Figure 1:</span> <strong>Deep Review editing workflow.</strong> Any reader can become a contributor by proposing a change through a pull request. In this example, the contributor opens an issue to discuss a manuscript modification. A maintainer and additional participant provide feedback, and the maintainer recommends creating a pull request to update the text. The contributor creates the pull request, it is reviewed by a maintainer and a participant, and the contributor updates the pull request in response. Once the pull request is approved, the maintainer merges the changes into the official version of the manuscript.</figcaption>
</figure>
<!-- Google Drawing View Link https://docs.google.com/drawings/d/17mst0Z1RMXegeGhM6SwQSJ4nhHgMMh5csyV4MwhsgAY/edit?usp=sharing -->
<p>The Deep Review <a href="https://github.com/greenelab/deep-review/issues/575">issue</a> and <a href="https://github.com/greenelab/deep-review/pull/638">pull request</a> on protein-protein interactions demonstrate this process in practice. A new contributor identified a relevant research topic that was missing from the review manuscript with examples of how the literature would be synthesized. A maintainer confirmed that this was a desirable topic and referred to related open issues. The contributor made the pull request, and two maintainers and another participant made recommendations. After four rounds of reviews and pull request edits, a maintainer merged the changes.</p>
<p>We found that this workflow was an effective compromise between fully unrestricted editing and a more heavily-structured approach that limited the authors or the sections they could edit. In addition, authors are associated with their commits, which makes it easy for contributors to receive credit for their work and helps prevent ghostwriting <span class="citation" data-cites="RK9sIADd">[<a href="#ref-RK9sIADd">15</a>]</span>. Figure <a href="#fig:contrib">2</a> and the GitHub <a href="https://github.com/greenelab/deep-review/graphs/contributors">contributors page</a> summarize all edits and commits from each author, providing aggregated information that is not available on other collaborative writing platforms. Because our writing process, like others backed by the open git version control system (including Overleaf and Authorea), tracks the complete commit history, it also enables detailed retrospective contribution analysis.</p>
<figure>
<img src="images/deep-review-contribution-ridge.svg" alt="Figure 2: Deep Review contributions by author over time. The total words added to the Deep Review by each author is plotted over time (final values in parentheses). These statistics were extracted from git commit diffs of the manuscript’s Markdown source. This figure reveals the composition of written contributions to the manuscript at every point in its history. Deep Review was initiated in August 2016, and the first complete manuscript was released as a preprint [16] in May 2017. While the article was under review, we continued to maintain the project and accepted new contributions. The preprint was updated in January 2018, and the article was accepted by the journal in March 2018 [5]." id="fig:contrib" style="width:100.0%" /><figcaption><span>Figure 2:</span> <strong>Deep Review contributions by author over time.</strong> The total words added to the Deep Review by each author is plotted over time (final values in parentheses). These statistics were extracted from git commit diffs of the manuscript’s Markdown source. This figure reveals the composition of written contributions to the manuscript at every point in its history. Deep Review was initiated in August 2016, and the first complete manuscript was released as a preprint <span class="citation" data-cites="tJKvnIaZ">[<a href="#ref-tJKvnIaZ">16</a>]</span> in May 2017. While the article was under review, we continued to maintain the project and accepted new contributions. The preprint was updated in January 2018, and the article was accepted by the journal in March 2018 <span class="citation" data-cites="PZMP42Ak">[<a href="#ref-PZMP42Ak">5</a>]</span>.</figcaption>
</figure>
<h2 id="manubot">Manubot</h2>
<p>We developed Manubot, a system for writing scholarly manuscripts via GitHub. With Manubot, manuscripts are written as plain-text Markdown files, which is well suited for version control using git. The Markdown standard itself provides limited yet crucial formatting syntax, including the ability to embed images and format text via bold, italics, hyperlinks, headers, inline code, codeblocks, blockquotes, and numbered or bulleted lists. In addition, Manubot relies on extensions from <a href="https://pandoc.org/MANUAL.html#pandocs-markdown">Pandoc Markdown</a> to enable citations, tables, captions, and equations specified using the popular TeX math syntax.</p>
<p>Manubot includes an additional layer of citation processing, currently unique to the system. All citations point to a standard identifier, for which Manubot automatically retrieves bibliographic metadata. Table <a href="#tbl:citations">1</a> presents the supported identifiers and example citations before and after Manubot processing. Authors can optionally define citation tags to provide short readable alternatives to the citation identifiers. Metadata is exported to <a href="http://citeproc-js.readthedocs.io/en/latest/csl-json/markup.html#items">Citation Style Language (CSL) JSON Items</a>, an open standard that is widely supported by reference managers <span class="citation" data-cites="9KfVIq3s K7WVgf8X">[<a href="#ref-9KfVIq3s">17</a>,<a href="#ref-K7WVgf8X">18</a>]</span>. In cases where automatic retrieval of metadata fails or produces incorrect references — which is most common for URL citations — users can manually provide the correct CSL JSON.</p>
<a name="tbl:citations"></a>
<table>
<caption><span>Table 1:</span> Citation types supported by Manubot. Authors may optionally map a named tag to one of the other supported identifier types. In this example, the tag <code>Avasthi2018_preprints</code> represents the DOI identifier <code>doi:10.7554/eLife.38532</code>. </caption>
<colgroup>
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
</colgroup>
<thead>
<tr class="header">
<th>Identifier</th>
<th>Metadata source</th>
<th>Example citation</th>
<th>Processed citation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Digital Object Identifier (DOI)</td>
<td>DOI <a href="https://citation.crosscite.org/docs.html">Content Negotiation</a></td>
<td><code>doi:10.1098/rsif.2017.0387</code></td>
<td><span class="citation" data-cites="PZMP42Ak">[<a href="#ref-PZMP42Ak">5</a>]</span></td>
</tr>
<tr class="even">
<td>PubMed Identifier (PMID)</td>
<td>NCBI’s <a href="https://www.ncbi.nlm.nih.gov/books/NBK25501/">E-utilities</a></td>
<td><code>pmid:25851694</code></td>
<td><span class="citation" data-cites="LfJGtB83">[<a href="#ref-LfJGtB83">6</a>]</span></td>
</tr>
<tr class="odd">
<td>PubMed Central Identifier (PMCID)</td>
<td>NCBI’s <a href="https://www.ncbi.nlm.nih.gov/pmc/tools/ctxp/">Citation Exporter</a></td>
<td><code>pmcid:PMC4719068</code></td>
<td><span class="citation" data-cites="12sHvZy1a">[<a href="#ref-12sHvZy1a">2</a>]</span></td>
</tr>
<tr class="even">
<td>arXiv identifier</td>
<td><a href="https://arxiv.org/help/api/index">arXiv API</a></td>
<td><code>arxiv:1502.04015v1</code></td>
<td><span class="citation" data-cites="Y2XyzLMc">[<a href="#ref-Y2XyzLMc">19</a>]</span></td>
</tr>
<tr class="odd">
<td>URL</td>
<td><a href="http://greycite.knowledgeblog.org/">Greycite</a> <span class="citation" data-cites="GKPtRdAw">[<a href="#ref-GKPtRdAw">20</a>]</span></td>
<td><code>url:https://lgatto.github.io/open-and-open/</code></td>
<td><span class="citation" data-cites="zBl3qgGT">[<a href="#ref-zBl3qgGT">21</a>]</span></td>
</tr>
<tr class="even">
<td>Tag</td>
<td>Source for tagged identifier</td>
<td><code>tag:Avasthi2018_preprints</code></td>
<td><span class="citation" data-cites="pqBLIXzp">[<a href="#ref-pqBLIXzp">9</a>]</span></td>
</tr>
</tbody>
</table>
<p>Manubot formats bibliographies according to a <a href="http://citationstyles.org/">CSL</a> specification. As a result, users can choose from thousands of existing CSL styles or use Manubot’s default style. Styles define how references are constructed from bibliographic metadata, controlling layout details such as the maximum number of authors to list per reference. Thousands of journals have <a href="http://editor.citationstyles.org/searchByName/">predefined styles</a>. As a result, adopting the specific bibliographic format required by a journal usually just requires specifying the style’s source URL in the Manubot configuration.</p>
<p>Manubot uses <a href="https://pandoc.org/">Pandoc</a> to convert manuscripts from Markdown to HTML, PDF, and optionally DOCX outputs. Pandoc supports conversion between additional formats — such as LaTeX, AsciiDoc, EPUB, and JATS — offering Manubot users broad interoperability. <a href="https://jats.nlm.nih.gov/">Journal Article Tag Suite</a> (JATS) is a standard XML format for scholarly articles that is used by publishers, archives, and text miners <span class="citation" data-cites="JU3KpeyB AAwqxolU bCyfIm6z">[<a href="#ref-JU3KpeyB">22</a>,<a href="#ref-AAwqxolU">23</a>,<a href="#ref-bCyfIm6z">24</a>]</span>. Going forward, we hope to integrate Manubot with the larger JATS ecosystem. For now however, the primary Manubot output is HTML intended to be viewed in a web browser.</p>
<p>Manubot performs continuous publication: every update to a manuscript’s source is automatically reflected in the online outputs. The approach uses continuous integration (CI) <span class="citation" data-cites="18w6XKsQO Qh7xTLwz lXvpQxeN">[<a href="#ref-18w6XKsQO">25</a>,<a href="#ref-Qh7xTLwz">26</a>,<a href="#ref-lXvpQxeN">27</a>]</span>, specifically <a href="https://travis-ci.org/">Travis CI</a> at the moment, to monitor for changes. When changes occur, the CI service generates an updated manuscript. If this process is error free, the CI service timestamps the manuscript and uploads the output files to the GitHub repository. Since the HTML manuscript is hosted using <a href="https://pages.github.com/">GitHub Pages</a>, the CI service automatically deploys the new manuscript version when it pushes the updated outputs to GitHub.</p>
<p>For this article, the source GitHub repository is <a href="https://github.com/greenelab/meta-review" class="uri">https://github.com/greenelab/meta-review</a>. When this repository changes, Travis CI <a href="https://travis-ci.org/greenelab/meta-review">rebuilds</a> the manuscript. If successful, the output is deployed back to GitHub (to dedicated <a href="https://github.com/greenelab/meta-review/tree/output"><code>output</code></a> and <a href="https://github.com/greenelab/meta-review/tree/gh-pages"><code>gh-pages</code></a> branches). As a result, <a href="https://greenelab.github.io/meta-review" class="uri">https://greenelab.github.io/meta-review</a> stays up to date with the latest HTML manuscript. Furthermore, versioned URLs, such as <a href="https://greenelab.github.io/meta-review/v/3ec66cd440e78dac33f4700f1cecbc4d95b20139/" class="uri">https://greenelab.github.io/meta-review/v/3ec66cd440e78dac33f4700f1cecbc4d95b20139/</a> <code>(TODO: update commit)</code>, provide access to previous manuscript versions.</p>
<p>The idea of the “priority of discovery” is important to science, and Vale and Hyman discuss the importance of both disclosure and validation <span class="citation" data-cites="vHuGhm4k">[<a href="#ref-vHuGhm4k">28</a>]</span>. In their framework, disclosure occurs when a scientific output is released to the world. However, for a manuscript that is shared as it is written, being able to establish priority could be challenging. We implemented support for <a href="https://opentimestamps.org/">OpenTimestamps</a> in Manubot to timestamp the HTML and PDF outputs on the Bitcoin blockchain. This procedure allows one to retrospectively prove that a manuscript version existed prior to its blockchain-verifiable timestamp <span class="citation" data-cites="Y2XyzLMc 6MR50hyY QBWMEuxW 6yyYojgV">[<a href="#ref-Y2XyzLMc">19</a>,<a href="#ref-6MR50hyY">29</a>,<a href="#ref-QBWMEuxW">30</a>,<a href="#ref-6yyYojgV">31</a>]</span>. For Manubot manuscripts, scientific precedence can now be indisputably established, and timestamps protect against attempts to rewrite a manuscript’s history. Such timestamping practices help ensure accurate histories, potentially alleviating certain authorship or priority disputes. Since all bitcoin transactions are competing for limited space on the blockchain, the fees required to send a single transaction can be high. OpenTimestamps avoids this fee by encoding many timestamps into a single Bitcoin transaction <span class="citation" data-cites="1DG704X8Q">[<a href="#ref-1DG704X8Q">32</a>]</span>, but there can be a lag of a few hours before the transaction is made. We judged this to be suitable for the purposes of scientific writing.</p>
<p>We designed Manubot to power the next generation of scholarly manuscript. Manubot transforms publication, making it permissionless, reproducible, free of charge, and largely open source. Manubot does rely on gratis services from two proprietary platforms: GitHub and Travis CI. Fortunately, lock-in to these services is minimal, and several substitutes already exist. One direction Manubot is working towards is end-to-end document reproducibility, where every figure or piece of data in a manuscript can be traced back to its origin <span class="citation" data-cites="sWD9uVuF">[<a href="#ref-sWD9uVuF">33</a>]</span>. Already, Manubot is well suited for preserving provenance. For example, figures can be specified using versioned URLs that refer to the code that created them. In addition, manuscripts can be templated, so that numerical values or tables get inserted directly from the repository that created them. An <a href="https://github.com/greenelab/manubot-rootstock">example repository</a> demonstrates Manubot’s features and serves as a template for users to write their own manuscript with Manubot.</p>
<p>Since its creation to facilitate the Deep Review, Manubot has been used to write a variety of scholarly documents. The <a href="https://github.com/greenelab/scihub-manuscript">Sci-Hub Coverage Study</a> — performed openly on GitHub from its inception — investigated Sci-Hub’s repository of pirated articles <span class="citation" data-cites="IhliSZDo">[<a href="#ref-IhliSZDo">34</a>]</span>. Sci-Hub <a href="https://github.com/greenelab/scihub-manuscript/issues/17">reviewed</a> the initial preprint from this study in a series of tweets, pointing out a major error in one of the analyses. Within hours, the authors used Markdown’s strikethrough formatting in Manubot to cross-out the errant sentences (<a href="https://github.com/greenelab/scihub-manuscript/commit/8fcd0cd665f6fb5f39bed7e26b940aa27d4770ba">commit</a>, <a href="https://greenelab.github.io/scihub-manuscript/v/8fcd0cd665f6fb5f39bed7e26b940aa27d4770ba/">versioned manuscript</a>), thereby alerting readers to the mistake and preventing further propagation of misinformation. A larger <a href="https://github.com/greenelab/scihub-manuscript/pull/19">set of revisions</a> that was merged a month later explained the error in more detail, and was included in a second version of the preprint. As such, continuous publishing via Manubot helped the authors address the error without delay, while retaining a public version history of the process. Of interest, the Sci-Hub Coverage Study preprint was the <a href="http://web.archive.org/web/20171221221858/http://www.prepubmed.org/top_preprints/">most viewed</a> 2017 <em>PeerJ Preprint</em>, while the Deep Review was the most viewed 2017 <em>bioRxiv</em> preprint <span class="citation" data-cites="9IrsqXRa">[<a href="#ref-9IrsqXRa">35</a>]</span>. Hence, in Manubot’s first year, two of the most popular preprints were written using its collaborative, open, and review-driven authoring process.</p>
<p>Additional research studies in progress are being authored using Manubot, including in the fields of <a href="https://vsmalladi.github.io/tfsee-manuscript/">genomics</a>, <a href="https://openclimatedata.github.io/global-emissions/">climate science</a>, and <a href="https://yt-project.github.io/yt-3.0-paper/">data visualization</a>. Manubot is also being used for documents beyond traditional journal publications, such as <a href="https://greenelab.github.io/manufund-2018/">grant proposals</a>, <a href="https://greenelab.github.io/czi-hca-report/">progress reports</a>, <a href="https://zietzm.github.io/Vagelos2017/">undergraduate research reports</a> <span class="citation" data-cites="15nwuvjrA">[<a href="#ref-15nwuvjrA">36</a>]</span>, <a href="https://slochower.github.io/synthetic-motor-literature/">literature reviews</a>, and lab notebooks. Finally, manuscripts written with other authoring systems have been successfully ported to Manubot, including the <a href="https://git.dhimmel.com/bitcoin-whitepaper/">Bitcoin Whitepaper</a> <span class="citation" data-cites="yte07Fnn">[<a href="#ref-yte07Fnn">37</a>]</span> and <a href="https://git.dhimmel.com/rephetio-manuscript/">Project Rephetio manuscript</a> <span class="citation" data-cites="O21tn8vf">[<a href="#ref-O21tn8vf">38</a>]</span>.</p>
<h2 id="authorship">Authorship</h2>
<p>To determine authorship for the Deep Review, we followed the International Committee of Medical Journal Editors (ICMJE) <a href="http://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html">guidelines</a> and used GitHub to track contributions. ICMJE recommends authors substantially contribute to, draft, approve, and agree to be accountable for the manuscript. We acknowledged other contributors who did not meet all four criteria, including contributors who provided text but did not review and approve the complete manuscript. Although these criteria provided a straightforward, equitable way to determine who would be an author, they did not produce a traditionally ordered author list. In biomedical journals, the convention is that the first and last authors made the most substantial contributions to the manuscript. This convention can be difficult to reconcile in a collaborative effort. Using git, we could quantify the number of commits each author made or the number of sentences an author wrote or edited, but these metrics discount intellectual contributions such as discussing primary literature and reviewing pull requests. However, there is no objective system to compare and weight the different types of contributions and produce an ordered author list.</p>
<p>To address this issue, we generalized the concept of “co-first” authorship, in which two or more authors are denoted as making equal contributions to a paper. We defined four types of contributions <span class="citation" data-cites="PZMP42Ak">[<a href="#ref-PZMP42Ak">5</a>]</span>, from major to minor, and reviewed the GitHub discussions and commits to assign authors to these categories. A randomized algorithm then arbitrarily ordered authors within each contribution category, and we combined the category-specific author lists to produce a traditional ordering. The randomization procedure was shared with the authors in advance (pre-registered) and run in a deterministic manner. Given the same author contributions, it always produced the same ordered author list. We annotated the author list to indicate that author order was partly randomized and emphasize that the order did not indicate one author contributed more than another from the same category.</p>
<h2 id="discussion">Discussion</h2>
<p>The Deep Review was not the first scholarly manuscript written online via an open collaborative process. Interestingly, the collaborative scholarly works (of which we’re aware) tend to have dealt with rapidly developing topics. In 2013, two dozen mathematicians created the 600-page “HoTT Book” on “Homotopy Type Theory”, writing collaboratively in LaTeX on GitHub <span class="citation" data-cites="RExXs8is MhC1nPDK">[<a href="#ref-RExXs8is">39</a>,<a href="#ref-MhC1nPDK">40</a>]</span>. Two technical books on cryptocurrency — <a href="https://github.com/bitcoinbook/bitcoinbook">Mastering Bitcoin</a> and <a href="https://github.com/ethereumbook/ethereumbook">Mastering Ethereum</a> — written on GitHub in asciidoc format have engaged hundreds of contributors. Both The HoTT Book and Mastering Bitcoin continue to be maintained years after their initial publication. A 2017 perspective on the future of peer review was written collaboratively on Overleaf, with contributions from 32 authors <span class="citation" data-cites="52Q1v5nS">[<a href="#ref-52Q1v5nS">41</a>]</span>. While debate was raging over tightening the default threshold for statistical significance, nearly 150 scientists contributed to a Google Doc discussion that was condensed into a traditional journal commentary <span class="citation" data-cites="82ZjWq3i HXpnCQu1">[<a href="#ref-82ZjWq3i">42</a>,<a href="#ref-HXpnCQu1">43</a>]</span>. The greatest success to date of open collaborative writing is arguably Wikipedia, whose English version contains over 5.5 million articles. Wikipedia scaled encyclopedias <a href="https://en.wikipedia.org/wiki/Wikipedia:Size_comparisons">far beyond</a> any privately-written alternative. As the examples above illustrate, open collaborative writing appears to scale scholarly manuscripts where diverse opinion and expertise are paramount beyond what would otherwise be possible.</p>
<p>Concepts for the future of scholarly publishing extend beyond collaborative writing <span class="citation" data-cites="WDvu1SAV ILhLpgTs">[<a href="#ref-WDvu1SAV">44</a>,<a href="#ref-ILhLpgTs">45</a>]</span>. Bookdown <span class="citation" data-cites="1BISa1RLr">[<a href="#ref-1BISa1RLr">46</a>]</span> and Pandoc Scholar <span class="citation" data-cites="17wKkS4DV">[<a href="#ref-17wKkS4DV">47</a>]</span> both extend traditional Markdown to better support publishing. Examples of continuous integration to automate manuscript generation include <a href="https://github.com/ewanmellor/gh-publisher">gh-publisher</a> and Continuous Publishing <span class="citation" data-cites="nqeDrtsc">[<a href="#ref-nqeDrtsc">48</a>]</span>, which was used to produce the book Opening Science <span class="citation" data-cites="ujrfOBM1">[<a href="#ref-ujrfOBM1">49</a>]</span>.</p>
<p>Many others have embraced open science principles and piloted open approaches toward drug discovery <span class="citation" data-cites="O21tn8vf HnZE3JiU">[<a href="#ref-O21tn8vf">38</a>,<a href="#ref-HnZE3JiU">50</a>]</span>, data management <span class="citation" data-cites="1A97a4UwU 6DjakjNS otY3SefD">[<a href="#ref-1A97a4UwU">51</a>,<a href="#ref-6DjakjNS">52</a>,<a href="#ref-otY3SefD">53</a>]</span>, and manuscript review <span class="citation" data-cites="nhLnUPJo">[<a href="#ref-nhLnUPJo">54</a>]</span>. Several of these open science efforts are GitHub-based like our collaborative writing process. The ReScience <span class="citation" data-cites="8o1nWux7">[<a href="#ref-8o1nWux7">55</a>]</span>, the Journal of Open Source Software <span class="citation" data-cites="xpw2aizK">[<a href="#ref-xpw2aizK">56</a>]</span>, and some other <a href="http://www.theoj.org/">Open Journals</a> rely on GitHub for peer review and hosting. GitHub is also increasingly used for resource curation <span class="citation" data-cites="vf9t7xMG">[<a href="#ref-vf9t7xMG">57</a>]</span>, and collaborative scholarly reviews combine literature curation with discussion and interpretation.</p>
<p>There are potential limitations of our GitHub-based approach. Because our review manuscript pertained to a computational topic, most of the authors had computational backgrounds, including previous experience with version control workflows and GitHub. In other disciplines, collaborative writing via GitHub and Manubot could present a steeper barrier to entry and deter participants. In addition, git carefully tracks all revisions to the manuscript text but not the surrounding conversations that take place through GitHub issues and pull requests. These discussions must be archived to ensure that important decisions about the manuscript are preserved and authors receive credit for intellectual contributions that are not directly reflected in the manuscript’s text. GitHub supports programmatic access to issues, pull requests, and reviews so tracking these conversations is feasible in the future.</p>
<p>In our open review paper, we established <a href="https://github.com/greenelab/deep-review/blob/v0.9-preprint/CONTRIBUTING.md">contributor guidelines</a> that discussed norms in the areas of text contribution, peer review, and authorship, which we identified in advance as potential areas of disagreement. Our contributor guidelines required verifiable participation: either directly attributable changes to the text or participation in the discussion on GitHub. We maintained our guidelines, even when a case arose where two authors had written text together but only one had directly attributable changes and participation. These guidelines did not discuss broader community norms that may have improved inclusiveness. It is also important to consider how the move to an open contribution model affects under-represented minority members of the scientific community <span class="citation" data-cites="zBl3qgGT">[<a href="#ref-zBl3qgGT">21</a>]</span>. Recent work has identified clear social norms and processes as helpful to maintaining a collaborative culture <span class="citation" data-cites="NuDPNceu">[<a href="#ref-NuDPNceu">58</a>]</span>. <!-- cited a previous version of doi:10.1080/08874417.2009.11645338 (see https://git.io/vdSSR) --> Conferences and open source projects have used codes of conduct to establish these norms <span class="citation" data-cites="HPKoE9m3 aRZRiJPk">[<a href="#ref-HPKoE9m3">59</a>,<a href="#ref-aRZRiJPk">60</a>]</span>. We would encourage the maintainers of similar projects to consider broader codes of conduct for project participants that establish on social as well as academic norms.</p>
<p>Open writing presents new opportunities for scholarly communication. Though it is still valuable to have versioned drafts of a review manuscript with digital identifiers, journal publication may not be the terminal endpoint for collaborative manuscripts. After releasing the first version of our collaborative review <span class="citation" data-cites="tJKvnIaZ">[<a href="#ref-tJKvnIaZ">16</a>]</span>, 14 new contributors have updated the manuscript (Figure <a href="#fig:contrib">2</a>) and existing authors continue to discuss new literature, <a href="https://github.com/greenelab/deep-review/">creating a living document</a>.</p>
<p>Our process represents an early step toward open massively collaborative reviews, and there are certainly aspects that can be improved. We invite the scientific community to adapt and build upon our experience and open software.</p>
<h2 id="acknowledgements">Acknowledgements</h2>
<p><code>TODO: deep review authors for support in testing this process</code> <code>TODO: manubot-rootstock contributors</code></p>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-HQfvK1OF">
<p>1. <strong>How open science helps researchers succeed</strong><br />
Erin C McKiernan, Philip E Bourne, C Titus Brown, Stuart Buck, Amye Kenall, Jennifer Lin, Damon McDougall, Brian A Nosek, Karthik Ram, Courtney K Soderberg, … Tal Yarkoni<br />
<em>eLife</em> (2016-07-07) <a href="https://doi.org/10.7554/elife.16800" class="uri">https://doi.org/10.7554/elife.16800</a></p>
</div>
<div id="ref-12sHvZy1a">
<p>2. <strong>Crowdsourcing in biomedicine: challenges and opportunities</strong><br />
Ritu Khare, Benjamin M. Good, Robert Leaman, Andrew I. Su, Zhiyong Lu<br />
<em>Briefings in Bioinformatics</em> (2016-01) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4719068/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4719068/</a></p>
</div>
<div id="ref-Rhm4AK0j">
<p>3. <strong>Ten Simple Rules for Writing a Literature Review</strong><br />
Marco Pautasso<br />
<em>PLoS Computational Biology</em> (2013-07-18) <a href="https://doi.org/10.1371/journal.pcbi.1003149" class="uri">https://doi.org/10.1371/journal.pcbi.1003149</a></p>
</div>
<div id="ref-16CgM2x0z">
<p>4. <strong>TechBlog: “Manubot” powers a crowdsourced “deep-learning” review</strong><br />
Jeffrey Perkel<br />
<em>Naturejobs</em> (2018-02-20) <a href="http://blogs.nature.com/naturejobs/2018/02/20/techblog-manubot-powers-a-crowdsourced-deep-learning-review/" class="uri">http://blogs.nature.com/naturejobs/2018/02/20/techblog-manubot-powers-a-crowdsourced-deep-learning-review/</a></p>
</div>
<div id="ref-PZMP42Ak">
<p>5. <strong>Opportunities and obstacles for deep learning in biology and medicine</strong><br />
Travers Ching, Daniel S. Himmelstein, Brett K. Beaulieu-Jones, Alexandr A. Kalinin, Brian T. Do, Gregory P. Way, Enrico Ferrero, Paul-Michael Agapow, Michael Zietz, Michael M. Hoffman, … Casey S. Greene<br />
<em>Journal of The Royal Society Interface</em> (2018-04) <a href="https://doi.org/10.1098/rsif.2017.0387" class="uri">https://doi.org/10.1098/rsif.2017.0387</a></p>
</div>
<div id="ref-LfJGtB83">
<p>6. <strong>Reviewing post-publication peer review.</strong><br />
Paul Knoepfler<br />
<em>Trends in genetics : TIG</em> (2015-04-04) <a href="https://www.ncbi.nlm.nih.gov/pubmed/25851694" class="uri">https://www.ncbi.nlm.nih.gov/pubmed/25851694</a></p>
</div>
<div id="ref-jYs2OUFW">
<p>7. <strong>A Stronger Post-Publication Culture Is Needed for Better Science</strong><br />
Hilda Bastian<br />
<em>PLoS Medicine</em> (2014-12-30) <a href="https://doi.org/10.1371/journal.pmed.1001772" class="uri">https://doi.org/10.1371/journal.pmed.1001772</a></p>
</div>
<div id="ref-HflJ6Hy5">
<p>8. <strong>Post-Publication Peer Review: Opening Up Scientific Conversation</strong><br />
Jane Hunter<br />
<em>Frontiers in Computational Neuroscience</em> (2012) <a href="https://doi.org/10.3389/fncom.2012.00063" class="uri">https://doi.org/10.3389/fncom.2012.00063</a></p>
</div>
<div id="ref-pqBLIXzp">
<p>9. <strong>Journal clubs in the time of preprints</strong><br />
Prachee Avasthi, Alice Soragni, Joshua N Bembenek<br />
<em>eLife</em> (2018-06-11) <a href="https://doi.org/10.7554/elife.38532" class="uri">https://doi.org/10.7554/elife.38532</a></p>
</div>
<div id="ref-uw5bep8P">
<p>10. <strong>Post-publication peer review, in all its guises, is here to stay</strong><br />
Michael Markie<br />
<em>Insights the UKSG journal</em> (2015-07-07) <a href="https://doi.org/10.1629/uksg.245" class="uri">https://doi.org/10.1629/uksg.245</a></p>
</div>
<div id="ref-AylLD9F8">
<p>11. <strong>Scientific writing: the online cooperative</strong><br />
Jeffrey M. Perkel<br />
<em>Nature</em> (2014-10-01) <a href="https://doi.org/10.1038/514127a" class="uri">https://doi.org/10.1038/514127a</a></p>
</div>
<div id="ref-PlcxShQU">
<p>12. <strong>A Quick Introduction to Version Control with Git and GitHub</strong><br />
John D. Blischak, Emily R. Davenport, Greg Wilson<br />
<em>PLOS Computational Biology</em> (2016-01-19) <a href="https://doi.org/10.1371/journal.pcbi.1004668" class="uri">https://doi.org/10.1371/journal.pcbi.1004668</a></p>
</div>
<div id="ref-kEX5dgzK">
<p>13. <strong>Ten Simple Rules for Taking Advantage of Git and GitHub</strong><br />
Yasset Perez-Riverol, Laurent Gatto, Rui Wang, Timo Sachsenberg, Julian Uszkoreit, Felipe da Veiga Leprevost, Christian Fufezan, Tobias Ternent, Stephen J. Eglen, Daniel S. Katz, … Juan Antonio Vizcaíno<br />
<em>PLOS Computational Biology</em> (2016-07-14) <a href="https://doi.org/10.1371/journal.pcbi.1004947" class="uri">https://doi.org/10.1371/journal.pcbi.1004947</a></p>
</div>
<div id="ref-opQBBK06">
<p>14. <strong>Opportunities And Obstacles For Deep Learning In Biology And Medicine</strong><br />
Johnny Israeli<br />
<em>Medium</em> (2017-05-31) <a href="https://medium.com/towards-data-science/opportunities-and-obstacles-for-deep-learning-in-biology-and-medicine-6ec914fe18c2" class="uri">https://medium.com/towards-data-science/opportunities-and-obstacles-for-deep-learning-in-biology-and-medicine-6ec914fe18c2</a></p>
</div>
<div id="ref-RK9sIADd">
<p>15. <strong>What Should Be Done To Tackle Ghostwriting in the Medical Literature?</strong><br />
Peter C Gøtzsche, Jerome P Kassirer, Karen L Woolley, Elizabeth Wager, Adam Jacobs, Art Gertel, Cindy Hamilton<br />
<em>PLoS Medicine</em> (2009-02-03) <a href="https://doi.org/10.1371/journal.pmed.1000023" class="uri">https://doi.org/10.1371/journal.pmed.1000023</a></p>
</div>
<div id="ref-tJKvnIaZ">
<p>16. <strong>Opportunities And Obstacles For Deep Learning In Biology And Medicine</strong><br />
Travers Ching, Daniel S. Himmelstein, Brett K. Beaulieu-Jones, Alexandr A. Kalinin, Brian T. Do, Gregory P. Way, Enrico Ferrero, Paul-Michael Agapow, Michael Zietz, Michael M Hoffman, … Casey S. Greene<br />
<em>Cold Spring Harbor Laboratory</em> (2017-05-28) <a href="https://doi.org/10.1101/142760" class="uri">https://doi.org/10.1101/142760</a></p>
</div>
<div id="ref-9KfVIq3s">
<p>17. <strong>Reference Management</strong><br />
Martin Fenner, Kaja Scheliga, Sönke Bartling<br />
<em>Opening Science</em> (2013-12-17) <a href="https://doi.org/10.1007/978-3-319-00026-8_8" class="uri">https://doi.org/10.1007/978-3-319-00026-8_8</a></p>
</div>
<div id="ref-K7WVgf8X">
<p>18. <strong>Comparison of Select Reference Management Tools</strong><br />
Yingting Zhang<br />
<em>Medical Reference Services Quarterly</em> (2012-01) <a href="https://doi.org/10.1080/02763869.2012.641841" class="uri">https://doi.org/10.1080/02763869.2012.641841</a></p>
</div>
<div id="ref-Y2XyzLMc">
<p>19. <strong>Decentralized Trusted Timestamping using the Crypto Currency Bitcoin</strong><br />
Bela Gipp, Norman Meuschke, André Gernandt<br />
<em>arXiv</em> (2015-02-13) <a href="https://arxiv.org/abs/1502.04015v1" class="uri">https://arxiv.org/abs/1502.04015v1</a></p>
</div>
<div id="ref-GKPtRdAw">
<p>20. <strong>Twenty-Five Shades of Greycite: Semantics for referencing and preservation</strong><br />
Phillip Lord, Lindsay Marshall<br />
<em>arXiv</em> (2013-04-26) <a href="https://arxiv.org/abs/1304.7151v1" class="uri">https://arxiv.org/abs/1304.7151v1</a></p>
</div>
<div id="ref-zBl3qgGT">
<p>21. <strong>Open science and open science</strong><br />
Laurent Gatto<br />
(2017-06-05) <a href="https://lgatto.github.io/open-and-open/" class="uri">https://lgatto.github.io/open-and-open/</a></p>
</div>
<div id="ref-JU3KpeyB">
<p>22. <strong>JATS: Journal Article Tag Suite, version 1.1</strong><br />
National Information Standards Organization<br />
(2015) <a href="http://www.niso.org/standards/z39-96-2015/" class="uri">http://www.niso.org/standards/z39-96-2015/</a></p>
</div>
<div id="ref-AAwqxolU">
<p>23. <strong>Journal Article Tag Suite 1.0: National Information Standards Organization standard of journal extensible markup language</strong><br />
Sun Huh<br />
<em>Science Editing</em> (2014-08-18) <a href="https://doi.org/10.6087/kcse.2014.1.99" class="uri">https://doi.org/10.6087/kcse.2014.1.99</a></p>
</div>
<div id="ref-bCyfIm6z">
<p>24. <strong>NISO Z39.96-201x, JATS: Journal Article Tag Suite</strong><br />
Mark H. Needleman<br />
<em>Serials Review</em> (2012-09) <a href="https://doi.org/10.1080/00987913.2012.10765464" class="uri">https://doi.org/10.1080/00987913.2012.10765464</a></p>
</div>
<div id="ref-18w6XKsQO">
<p>25. <strong>Collaborative software development made easy</strong><br />
Andrew Silver<br />
<em>Nature</em> (2017-10-04) <a href="https://doi.org/10.1038/550143a" class="uri">https://doi.org/10.1038/550143a</a></p>
</div>
<div id="ref-Qh7xTLwz">
<p>26. <strong>Reproducibility of computational workflows is automated using continuous analysis</strong><br />
Brett K Beaulieu-Jones, Casey S Greene<br />
<em>Nature Biotechnology</em> (2017-03-13) <a href="https://doi.org/10.1038/nbt.3780" class="uri">https://doi.org/10.1038/nbt.3780</a></p>
</div>
<div id="ref-lXvpQxeN">
<p>27. <strong>Developing a modern data workflow for living data</strong><br />
Glenda M Yenni, Erica M Christensen, Ellen K Bledsoe, Sarah R Supp, Renata M Diaz, Ethan P White, SK Morgan Ernest<br />
<em>Cold Spring Harbor Laboratory</em> (2018-06-12) <a href="https://doi.org/10.1101/344804" class="uri">https://doi.org/10.1101/344804</a></p>
</div>
<div id="ref-vHuGhm4k">
<p>28. <strong>Priority of discovery in the life sciences</strong><br />
Ronald D Vale, Anthony A Hyman<br />
<em>eLife</em> (2016-06-16) <a href="https://doi.org/10.7554/elife.16931" class="uri">https://doi.org/10.7554/elife.16931</a></p>
</div>
<div id="ref-6MR50hyY">
<p>29. <strong>The Grey Literature — Proof of prespecified endpoints in medical research with the bitcoin blockchain</strong><br />
Benjamin Gregory Carlisle<br />
(2014-08-25) <a href="https://www.bgcarlisle.com/blog/2014/08/25/proof-of-prespecified-endpoints-in-medical-research-with-the-bitcoin-blockchain/" class="uri">https://www.bgcarlisle.com/blog/2014/08/25/proof-of-prespecified-endpoints-in-medical-research-with-the-bitcoin-blockchain/</a></p>
</div>
<div id="ref-QBWMEuxW">
<p>30. <strong>The most interesting case of scientific irreproducibility?</strong><br />
Daniel Himmelstein<br />
<em>Satoshi Village</em> (2017-03-08) <a href="http://blog.dhimmel.com/irreproducible-timestamps/" class="uri">http://blog.dhimmel.com/irreproducible-timestamps/</a></p>
</div>
<div id="ref-6yyYojgV">
<p>31. <strong>Bitcoin: A Peer-to-Peer Electronic Cash System</strong><br />
Satoshi Nakamoto<br />
(2017-09-20) <a href="http://git.dhimmel.com/bitcoin-whitepaper/" class="uri">http://git.dhimmel.com/bitcoin-whitepaper/</a></p>
</div>
<div id="ref-1DG704X8Q">
<p>32. <strong>OpenTimestamps: Scalable, Trustless, Distributed Timestamping with Bitcoin</strong><em>Peter Todd</em> (2018-02-10) <a href="https://petertodd.org/2016/opentimestamps-announcement" class="uri">https://petertodd.org/2016/opentimestamps-announcement</a></p>
</div>
<div id="ref-sWD9uVuF">
<p>33. <strong>eLife supports development of open technology stack for publishing reproducible manuscripts online</strong><br />
Emily Packer<br />
(2017-09-07) <a href="https://elifesciences.org/for-the-press/e6038800/elife-supports-development-of-open-technology-stack-for-publishing-reproducible-manuscripts-online" class="uri">https://elifesciences.org/for-the-press/e6038800/elife-supports-development-of-open-technology-stack-for-publishing-reproducible-manuscripts-online</a></p>
</div>
<div id="ref-IhliSZDo">
<p>34. <strong>Sci-Hub provides access to nearly all scholarly literature</strong><br />
Daniel S Himmelstein, Ariel Rodriguez Romero, Jacob G Levernier, Thomas Anthony Munro, Stephen Reid McLaughlin, Bastian Greshake Tzovaras, Casey S Greene<br />
<em>eLife</em> (2018-03-01) <a href="https://doi.org/10.7554/elife.32822" class="uri">https://doi.org/10.7554/elife.32822</a></p>
</div>
<div id="ref-9IrsqXRa">
<p>35. <strong>2017 in news: The science events that shaped the year</strong><br />
Ewen Callaway, Davide Castelvecchi, David Cyranoski, Elizabeth Gibney, Heidi Ledford, Jane J. Lee, Lauren Morello, Nicky Phillips, Quirin Schiermeier, Jeff Tollefson, … Alexandra Witze<br />
<em>Nature</em> (2017-12-21) <a href="https://doi.org/10.1038/d41586-017-08493-x" class="uri">https://doi.org/10.1038/d41586-017-08493-x</a></p>
</div>
<div id="ref-15nwuvjrA">
<p>36. <strong>Vagelos Report Summer 2017</strong><br />
Michael Zietz<br />
<em>Figshare</em> (2017) <a href="https://doi.org/10.6084/m9.figshare.5346577" class="uri">https://doi.org/10.6084/m9.figshare.5346577</a></p>
</div>
<div id="ref-yte07Fnn">
<p>37. <strong>How I used the Manubot to reproduce the Bitcoin Whitepaper</strong><br />
Daniel Himmelstein<br />
<em>Steem</em> (2017-09-20) <a href="https://steemit.com/manubot/@dhimmel/how-i-used-the-manubot-to-reproduce-the-bitcoin-whitepaper" class="uri">https://steemit.com/manubot/@dhimmel/how-i-used-the-manubot-to-reproduce-the-bitcoin-whitepaper</a></p>
</div>
<div id="ref-O21tn8vf">
<p>38. <strong>Systematic integration of biomedical knowledge prioritizes drugs for repurposing</strong><br />
Daniel Scott Himmelstein, Antoine Lizee, Christine Hessler, Leo Brueggeman, Sabrina L Chen, Dexter Hadley, Ari Green, Pouya Khankhanian, Sergio E Baranzini<br />
<em>eLife</em> (2017-09-22) <a href="https://doi.org/10.7554/elife.26726" class="uri">https://doi.org/10.7554/elife.26726</a></p>
</div>
<div id="ref-RExXs8is">
<p>39. <strong>Homotopy Type Theory: Univalent Foundations of Mathematics</strong><br />
The Univalent Foundations Program<br />
<em>Institute for Advanced Study</em> (2013) <a href="https://homotopytypetheory.org/book/" class="uri">https://homotopytypetheory.org/book/</a></p>
</div>
<div id="ref-MhC1nPDK">
<p>40. <strong>The HoTT book</strong><br />
Andrej Bauer<br />
<em>Mathematics and Computation</em> (2013-06-20) <a href="http://math.andrej.com/2013/06/20/the-hott-book/" class="uri">http://math.andrej.com/2013/06/20/the-hott-book/</a></p>
</div>
<div id="ref-52Q1v5nS">
<p>41. <strong>A multi-disciplinary perspective on emergent and future innovations in peer review</strong><br />
Jonathan P. Tennant, Jonathan M. Dugan, Daniel Graziotin, Damien C. Jacques, François Waldner, Daniel Mietchen, Yehia Elkhatib, Lauren B. Collister, Christina K. Pikas, Tom Crick, … Julien Colomb<br />
<em>F1000Research</em> (2017-11-01) <a href="https://doi.org/10.12688/f1000research.12037.2" class="uri">https://doi.org/10.12688/f1000research.12037.2</a></p>
</div>
<div id="ref-82ZjWq3i">
<p>42. <strong>Nearly 100 scientists spent 2 months on Google Docs to redefine the p-value. Here’s what they came up with</strong><br />
Jop Vrieze<br />
<em>Science</em> (2018-01-18) <a href="https://doi.org/10.1126/science.aat0471" class="uri">https://doi.org/10.1126/science.aat0471</a></p>
</div>
<div id="ref-HXpnCQu1">
<p>43. <strong>Justify your alpha</strong><br />
Daniel Lakens, Federico G. Adolfi, Casper J. Albers, Farid Anvari, Matthew A. J. Apps, Shlomo E. Argamon, Thom Baguley, Raymond B. Becker, Stephen D. Benning, Daniel E. Bradford, … Rolf A. Zwaan<br />
<em>Nature Human Behaviour</em> (2018-02-26) <a href="https://doi.org/10.1038/s41562-018-0311-x" class="uri">https://doi.org/10.1038/s41562-018-0311-x</a></p>
</div>
<div id="ref-WDvu1SAV">
<p>44. <strong>The arXiv of the future will not look like the arXiv</strong><br />
Alberto Pepe, Matteo Cantiello, Josh Nicholson<br />
<em>Authorea</em> <a href="https://doi.org/10.22541/au.149693987.70506124" class="uri">https://doi.org/10.22541/au.149693987.70506124</a></p>
</div>
<div id="ref-ILhLpgTs">
<p>45. <strong>TechBlog: C. Titus Brown: Predicting the paper of the future</strong><br />
C. Titus Brown<br />
<em>Naturejobs</em> (2017-06-01) <a href="http://blogs.nature.com/naturejobs/2017/06/01/techblog-c-titus-brown-predicting-the-paper-of-the-future/" class="uri">http://blogs.nature.com/naturejobs/2017/06/01/techblog-c-titus-brown-predicting-the-paper-of-the-future/</a></p>
</div>
<div id="ref-1BISa1RLr">
<p>46. <strong>bookdown</strong><br />
Yihui Xie<br />
<em>Chapman &amp;Hall/CRC The R Series</em> (2016-12-21) <a href="https://doi.org/10.1201/9781315204963" class="uri">https://doi.org/10.1201/9781315204963</a></p>
</div>
<div id="ref-17wKkS4DV">
<p>47. <strong>Formatting Open Science: agilely creating multiple document formats for academic manuscripts with Pandoc Scholar</strong><br />
Albert Krewinkel, Robert Winkler<br />
<em>PeerJ Computer Science</em> (2017-05-08) <a href="https://doi.org/10.7717/peerj-cs.112" class="uri">https://doi.org/10.7717/peerj-cs.112</a></p>
</div>
<div id="ref-nqeDrtsc">
<p>48. <strong>Continuous Publishing</strong><br />
Martin Fenner<br />
<em>Gobbledygook</em> (2014-03-10) <a href="http://blog.martinfenner.org/2014/03/10/continuous-publishing/" class="uri">http://blog.martinfenner.org/2014/03/10/continuous-publishing/</a></p>
</div>
<div id="ref-ujrfOBM1">
<p>49. <strong>Opening Science</strong><em>Springer International Publishing</em> (2014) <a href="https://doi.org/10.1007/978-3-319-00026-8" class="uri">https://doi.org/10.1007/978-3-319-00026-8</a></p>
</div>
<div id="ref-HnZE3JiU">
<p>50. <strong>An open source pharma roadmap</strong><br />
Manica Balasegaram, Peter Kolb, John McKew, Jaykumar Menon, Piero Olliaro, Tomasz Sablinski, Zakir Thomas, Matthew H. Todd, Els Torreele, John Wilbanks<br />
<em>PLOS Medicine</em> (2017-04-18) <a href="https://doi.org/10.1371/journal.pmed.1002276" class="uri">https://doi.org/10.1371/journal.pmed.1002276</a></p>
</div>
<div id="ref-1A97a4UwU">
<p>51. <strong>The Open Knowledge Foundation: Open Data Means Better Science</strong><br />
Jennifer C. Molloy<br />
<em>PLoS Biology</em> (2011-12-06) <a href="https://doi.org/10.1371/journal.pbio.1001195" class="uri">https://doi.org/10.1371/journal.pbio.1001195</a></p>
</div>
<div id="ref-6DjakjNS">
<p>52. <strong>The FAIR Guiding Principles for scientific data management and stewardship</strong><br />
Mark D. Wilkinson, Michel Dumontier, IJsbrand Jan Aalbersberg, Gabrielle Appleton, Myles Axton, Arie Baak, Niklas Blomberg, Jan-Willem Boiten, Luiz Bonino da Silva Santos, Philip E. Bourne, … Barend Mons<br />
<em>Scientific Data</em> (2016-03-15) <a href="https://doi.org/10.1038/sdata.2016.18" class="uri">https://doi.org/10.1038/sdata.2016.18</a></p>
</div>
<div id="ref-otY3SefD">
<p>53. <strong>First, design for data sharing</strong><br />
John Wilbanks, Stephen H Friend<br />
<em>Nature Biotechnology</em> (2016-04) <a href="https://doi.org/10.1038/nbt.3516" class="uri">https://doi.org/10.1038/nbt.3516</a></p>
</div>
<div id="ref-nhLnUPJo">
<p>54. <strong>A multi-disciplinary perspective on emergent and future innovations in peer review</strong><br />
Jonathan P. Tennant, Jonathan M. Dugan, Daniel Graziotin, Damien C. Jacques, François Waldner, Daniel Mietchen, Yehia Elkhatib, Lauren B. Collister, Christina K. Pikas, Tom Crick, … Julien Colomb<br />
<em>F1000Research</em> (2017-07-20) <a href="https://doi.org/10.12688/f1000research.12037.1" class="uri">https://doi.org/10.12688/f1000research.12037.1</a></p>
</div>
<div id="ref-8o1nWux7">
<p>55. <strong>Sustainable computational science: the ReScience initiative</strong><br />
Nicolas P. Rougier, Konrad Hinsen, Frédéric Alexandre, Thomas Arildsen, Lorena A. Barba, Fabien C.Y. Benureau, C. Titus Brown, Pierre de Buyl, Ozan Caglayan, Andrew P. Davison, … Tiziano Zito<br />
<em>PeerJ Computer Science</em> (2017-12-18) <a href="https://doi.org/10.7717/peerj-cs.142" class="uri">https://doi.org/10.7717/peerj-cs.142</a></p>
</div>
<div id="ref-xpw2aizK">
<p>56. <strong>Journal of Open Source Software (JOSS): design and first-year review</strong><br />
Arfon M. Smith, Kyle E. Niemeyer, Daniel S. Katz, Lorena A. Barba, George Githinji, Melissa Gymrek, Kathryn D. Huff, Christopher R. Madan, Abigail Cabunoc Mayes, Kevin M. Moerman, … Jacob T. Vanderplas<br />
<em>PeerJ Computer Science</em> (2018-02-12) <a href="https://doi.org/10.7717/peerj-cs.147" class="uri">https://doi.org/10.7717/peerj-cs.147</a></p>
</div>
<div id="ref-vf9t7xMG">
<p>57. <strong>The appropriation of GitHub for curation</strong><br />
Yu Wu, Na Wang, Jessica Kropczynski, John M. Carroll<br />
<em>PeerJ Computer Science</em> (2017-10-09) <a href="https://doi.org/10.7717/peerj-cs.134" class="uri">https://doi.org/10.7717/peerj-cs.134</a></p>
</div>
<div id="ref-NuDPNceu">
<p>58. <strong>Innovating Collaborative Content Creation: The Role of Altruism and Wiki Technology</strong><br />
Christian Wagner, Pattarawan Prasarnphanich<br />
<em>2007 40th Annual Hawaii International Conference on System Sciences (HICSS’07)</em> (2007) <a href="https://doi.org/10.1109/hicss.2007.277" class="uri">https://doi.org/10.1109/hicss.2007.277</a></p>
</div>
<div id="ref-HPKoE9m3">
<p>59. <strong>Code of conduct evaluations</strong><em>Geek Feminism Wiki</em> (2017-06-13) <a href="http://geekfeminism.wikia.com/wiki/Code_of_conduct_evaluations?oldid" class="uri">http://geekfeminism.wikia.com/wiki/Code_of_conduct_evaluations?oldid</a></p>
</div>
<div id="ref-aRZRiJPk">
<p>60. <strong>Contributor Covenant: A Code of Conduct for Open Source Projects</strong><br />
Coraline Ada Ehmke<br />
(2014) <a href="https://www.contributor-covenant.org/" class="uri">https://www.contributor-covenant.org/</a></p>
</div>
</div>
<script>
// AnchorJS minified version below.
// Source https://github.com/bryanbraun/anchorjs/blob/064abdd0987f305933ec4982af6d0c1cf2fd0814/anchor.js

/**
 * AnchorJS - v4.0.0 - 2017-06-02
 * https://github.com/bryanbraun/anchorjs
 * Copyright (c) 2017 Bryan Braun; Licensed MIT
 */
!function(A,e){"use strict";"function"==typeof define&&define.amd?define([],e):"object"==typeof module&&module.exports?module.exports=e():(A.AnchorJS=e(),A.anchors=new A.AnchorJS)}(this,function(){"use strict";function A(A){function e(A){A.icon=A.hasOwnProperty("icon")?A.icon:"",A.visible=A.hasOwnProperty("visible")?A.visible:"hover",A.placement=A.hasOwnProperty("placement")?A.placement:"right",A.class=A.hasOwnProperty("class")?A.class:"",A.truncate=A.hasOwnProperty("truncate")?Math.floor(A.truncate):64}function t(A){var e;if("string"==typeof A||A instanceof String)e=[].slice.call(document.querySelectorAll(A));else{if(!(Array.isArray(A)||A instanceof NodeList))throw new Error("The selector provided to AnchorJS was invalid.");e=[].slice.call(A)}return e}function n(){if(null===document.head.querySelector("style.anchorjs")){var A,e=document.createElement("style");e.className="anchorjs",e.appendChild(document.createTextNode("")),void 0===(A=document.head.querySelector('[rel="stylesheet"], style'))?document.head.appendChild(e):document.head.insertBefore(e,A),e.sheet.insertRule(" .anchorjs-link {   opacity: 0;   text-decoration: none;   -webkit-font-smoothing: antialiased;   -moz-osx-font-smoothing: grayscale; }",e.sheet.cssRules.length),e.sheet.insertRule(" *:hover > .anchorjs-link, .anchorjs-link:focus  {   opacity: 1; }",e.sheet.cssRules.length),e.sheet.insertRule(" [data-anchorjs-icon]::after {   content: attr(data-anchorjs-icon); }",e.sheet.cssRules.length),e.sheet.insertRule(' @font-face {   font-family: "anchorjs-icons";   src: url(data:n/a;base64,AAEAAAALAIAAAwAwT1MvMg8yG2cAAAE4AAAAYGNtYXDp3gC3AAABpAAAAExnYXNwAAAAEAAAA9wAAAAIZ2x5ZlQCcfwAAAH4AAABCGhlYWQHFvHyAAAAvAAAADZoaGVhBnACFwAAAPQAAAAkaG10eASAADEAAAGYAAAADGxvY2EACACEAAAB8AAAAAhtYXhwAAYAVwAAARgAAAAgbmFtZQGOH9cAAAMAAAAAunBvc3QAAwAAAAADvAAAACAAAQAAAAEAAHzE2p9fDzz1AAkEAAAAAADRecUWAAAAANQA6R8AAAAAAoACwAAAAAgAAgAAAAAAAAABAAADwP/AAAACgAAA/9MCrQABAAAAAAAAAAAAAAAAAAAAAwABAAAAAwBVAAIAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAMCQAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAg//0DwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAAIAAAACgAAxAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADAAAAAIAAgAAgAAACDpy//9//8AAAAg6cv//f///+EWNwADAAEAAAAAAAAAAAAAAAAACACEAAEAAAAAAAAAAAAAAAAxAAACAAQARAKAAsAAKwBUAAABIiYnJjQ3NzY2MzIWFxYUBwcGIicmNDc3NjQnJiYjIgYHBwYUFxYUBwYGIwciJicmNDc3NjIXFhQHBwYUFxYWMzI2Nzc2NCcmNDc2MhcWFAcHBgYjARQGDAUtLXoWOR8fORYtLTgKGwoKCjgaGg0gEhIgDXoaGgkJBQwHdR85Fi0tOAobCgoKOBoaDSASEiANehoaCQkKGwotLXoWOR8BMwUFLYEuehYXFxYugC44CQkKGwo4GkoaDQ0NDXoaShoKGwoFBe8XFi6ALjgJCQobCjgaShoNDQ0NehpKGgobCgoKLYEuehYXAAAADACWAAEAAAAAAAEACAAAAAEAAAAAAAIAAwAIAAEAAAAAAAMACAAAAAEAAAAAAAQACAAAAAEAAAAAAAUAAQALAAEAAAAAAAYACAAAAAMAAQQJAAEAEAAMAAMAAQQJAAIABgAcAAMAAQQJAAMAEAAMAAMAAQQJAAQAEAAMAAMAAQQJAAUAAgAiAAMAAQQJAAYAEAAMYW5jaG9yanM0MDBAAGEAbgBjAGgAbwByAGoAcwA0ADAAMABAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAH//wAP) format("truetype"); }',e.sheet.cssRules.length)}}this.options=A||{},this.elements=[],e(this.options),this.isTouchDevice=function(){return!!("ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch)},this.add=function(A){var i,o,s,c,r,a,h,l,u,d,f,g,p=[];if(e(this.options),"touch"===(g=this.options.visible)&&(g=this.isTouchDevice()?"always":"hover"),A||(A="h2, h3, h4, h5, h6"),0===(i=t(A)).length)return this;for(n(),o=document.querySelectorAll("[id]"),s=[].map.call(o,function(A){return A.id}),r=0;r<i.length;r++)if(this.hasAnchorJSLink(i[r]))p.push(r);else{if(i[r].hasAttribute("id"))c=i[r].getAttribute("id");else if(i[r].hasAttribute("data-anchor-id"))c=i[r].getAttribute("data-anchor-id");else{u=l=this.urlify(i[r].textContent),h=0;do{void 0!==a&&(u=l+"-"+h),a=s.indexOf(u),h+=1}while(-1!==a);a=void 0,s.push(u),i[r].setAttribute("id",u),c=u}d=c.replace(/-/g," "),(f=document.createElement("a")).className="anchorjs-link "+this.options.class,f.href="#"+c,f.setAttribute("aria-label","Anchor link for: "+d),f.setAttribute("data-anchorjs-icon",this.options.icon),"always"===g&&(f.style.opacity="1"),""===this.options.icon&&(f.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(f.style.lineHeight="inherit")),"left"===this.options.placement?(f.style.position="absolute",f.style.marginLeft="-1em",f.style.paddingRight="0.5em",i[r].insertBefore(f,i[r].firstChild)):(f.style.paddingLeft="0.375em",i[r].appendChild(f))}for(r=0;r<p.length;r++)i.splice(p[r]-r,1);return this.elements=this.elements.concat(i),this},this.remove=function(A){for(var e,n,i=t(A),o=0;o<i.length;o++)(n=i[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(i[o]))&&this.elements.splice(e,1),i[o].removeChild(n));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}return A});

// Enable links for selected headers
var anchors = new AnchorJS();
anchors.add("h2, h3, h4")
</script>
<!-- Enable Hypothesis annotation. https://web.hypothes.is/ -->
<script src="https://hypothes.is/embed.js" async></script>
</body>
</html>
